<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>“Single-cell” spatial transcriptomics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="chapter9_spatial_files/libs/clipboard/clipboard.min.js"></script>
<script src="chapter9_spatial_files/libs/quarto-html/quarto.js"></script>
<script src="chapter9_spatial_files/libs/quarto-html/popper.min.js"></script>
<script src="chapter9_spatial_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="chapter9_spatial_files/libs/quarto-html/anchor.min.js"></script>
<link href="chapter9_spatial_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="chapter9_spatial_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="chapter9_spatial_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="chapter9_spatial_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="chapter9_spatial_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


<link rel="stylesheet" href="styles.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#the-broad-categorization-of-spatial-transcriptomic-technologies" id="toc-the-broad-categorization-of-spatial-transcriptomic-technologies" class="nav-link active" data-scroll-target="#the-broad-categorization-of-spatial-transcriptomic-technologies"><span class="header-section-number">1</span> The broad categorization of spatial transcriptomic technologies</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">“Single-cell” spatial transcriptomics</h1>
<p class="subtitle lead">????</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Single-cell spatial transcriptomics is an emerging technology that enables the measurement of gene expression while preserving the spatial organization of cells within a tissue. See <a href="#fig-lego">Figure&nbsp;1</a> for a useful schematic I like. Unlike traditional single-cell RNA sequencing, which dissociates cells and loses spatial context, spatial transcriptomics allows researchers to analyze gene expression in relation to cellular neighborhoods, tissue architecture, and microenvironments, see <a href="#fig-st-ngs">Figure&nbsp;2</a>.</p>
<div class="block">
<div id="fig-lego" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/lego.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;1: From https://www.mscience.com.au/supplier/10x-genomics/.</figcaption>
</figure>
</div>
</div>
<div class="block">
<div id="fig-st-ngs" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/st-ngs.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;2: From <span class="citation" data-cites="kim2023single">Kim, Choi, and Lee (<a href="#ref-kim2023single" role="doc-biblioref">2023</a>)</span>.</figcaption>
</figure>
</div>
</div>
<p>The spatial arrangement of cells is fundamental to understanding cell biology, as cellular functions are often influenced by their local microenvironment:</p>
<ul>
<li><strong>Cancer</strong>: The tumor microenvironment plays a crucial role in disease progression, immune evasion, and therapeutic response. Understanding where specific cell populations reside within a tumor and how they interact with stromal and immune cells can reveal mechanisms of resistance and potential therapeutic targets.</li>
<li><strong>Developmental biology</strong>: Spatial positioning dictates lineage specification and organ formation, where disruptions in cell placement can lead to congenital defects. We can also overlay the gene expression at each spatial location with other imaging technologies or our current understanding of the structure of an organ, see <a href="#fig-spatial-anatomy">Figure&nbsp;3</a>. This can help us learn how the cells in different regions of an organ have different functions, even if they are all the “same cell type.”</li>
</ul>
<div class="block">
<div id="fig-spatial-anatomy" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial-anatomy.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;3: From <span class="citation" data-cites="vanrobaeys2023spatial">Vanrobaeys et al. (<a href="#ref-vanrobaeys2023spatial" role="doc-biblioref">2023</a>)</span>.</figcaption>
</figure>
</div>
</div>
<ul>
<li><strong>Disease</strong>: In neurodegenerative diseases such as Alzheimer’s, the spatial distribution of microglia and astrocytes based on their (spatial) proximity to pathological hallmarks like amyloid plaques and tau tangles provides critical insights into disease mechanisms.</li>
</ul>
<p>As one concrete example of how spatial transcriptomics advances cell biology beyond what other sequencing technologies offers: spatial transcriptomics is particularly valuable for studying cell-cell communication, as it enables the identification of ligand-receptor interactions that mediate signaling between neighboring cells. By integrating spatial data with single-cell transcriptomics, researchers can infer functional relationships between different cell types, uncovering regulatory networks that drive biological processes.</p>
<p><strong>Note</strong>: Spatial information in cell biology has been studied for decades, long before single-cell sequencing became commercially feasible. Techniques like fluorescence in situ hybridization (FISH) have been used to visualize the spatial localization of specific RNA molecules within cells, providing crucial insights into gene expression patterns. Microscopy-based methods, including confocal and super-resolution imaging, have also been instrumental in understanding cellular structures and interactions. The key novelty of modern spatial transcriptomics is its ability to scale these analyses – rather than measuring just a few genes at a time, we can now capture the spatial expression patterns of hundreds to thousands of genes simultaneously, enabling a much more comprehensive view of cellular organization and function.</p>
<section id="the-broad-categorization-of-spatial-transcriptomic-technologies" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="the-broad-categorization-of-spatial-transcriptomic-technologies"><span class="header-section-number">1</span> The broad categorization of spatial transcriptomic technologies</h2>
<p>Spatial transcriptomic technologies can be broadly categorized into two main approaches: spot-based methods and imaging-based methods, see <a href="#fig-spatial-resolution1">Figure&nbsp;4</a> and <a href="#fig-spatial-resolution2">Figure&nbsp;5</a>.</p>
<div class="block">
<div id="fig-spatial-resolution1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial-resolution.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;4: (Left) From https://hutchdatascience.org/Choosing_Genomics_Tools/spatial-transcriptomics-1.html.</figcaption>
</figure>
</div>
</div>
<div class="block">
<div id="fig-spatial-resolution2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial-resolution2.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;5: (Right) From https://geneviatechnologies.com/bioinformatics-analyses/spatial-transcriptomic-data-analysis/</figcaption>
</figure>
</div>
</div>
<ul>
<li><p><strong>Spot-based technologies</strong> utilize sequencing-based approaches to capture gene expression across a tissue section. These methods typically involve placing spatially barcoded capture spots on a slide, allowing RNA molecules from the tissue to be sequenced while retaining their spatial information. Spot-based techniques offer broad coverage across the entire transcriptome, making them well-suited for large tissue sections and unbiased gene discovery. However, they have relatively low spatial resolution due to the size of the capture spots and may lack sufficient depth for detecting lowly expressed but biologically significant genes.</p></li>
<li><p><strong>Imaging-based technologies</strong> rely on multiplexed imaging methods to visualize gene expression directly within tissues. These approaches build upon fluorescence in situ hybridization (FISH) and other imaging techniques, enabling subcellular resolution of RNA molecules. By using sequential hybridization cycles with fluorescent probes, imaging-based methods can measure the expression of dozens to hundreds of preselected genes. While this high-resolution approach provides detailed spatial maps at the single-molecule level, it requires prior selection of target genes and computationally intensive methods to delineate cell boundaries from dense tissue images.</p></li>
</ul>
<p>See <a href="#fig-spatial-comparison">Figure&nbsp;6</a>, <a href="#fig-spatial-comparison3">Figure&nbsp;7</a>, and <a href="#fig-spatial-sidebyside">Figure&nbsp;8</a> for examples of what these datasets look like, and how spot- and imaging-based spatial transcriptomics could yield quite dramatically different datasets.</p>
<div class="block">
<div id="fig-spatial-comparison" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial_comparison.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;6: (Top) From https://www.trillium.de/en/journals/trillium-pathology/archive/2024/tp-1/2024/ai-in-pathology/integration-of-spatially-resolved-transcriptomics-into-pathological-research-opportunities-and-challenges.html. (Bottom) From <span class="citation" data-cites="figiel2024clinical">Figiel et al. (<a href="#ref-figiel2024clinical" role="doc-biblioref">2024</a>)</span>.</figcaption>
</figure>
</div>
</div>
<div class="block">
<div id="fig-spatial-comparison3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial_comparison3.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;7: From <span class="citation" data-cites="longo2021integrating">Longo et al. (<a href="#ref-longo2021integrating" role="doc-biblioref">2021</a>)</span>.</figcaption>
</figure>
</div>
</div>
<div class="block">
<div id="fig-spatial-sidebyside" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial-sidebyside.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;8: From <span class="citation" data-cites="rademacher2024comparison">Rademacher et al. (<a href="#ref-rademacher2024comparison" role="doc-biblioref">2024</a>)</span>.</figcaption>
</figure>
</div>
</div>
<p>See <a href="#fig-spatial-table">Figure&nbsp;9</a> for a table of many possible technologies to perform spatial transcriptomics.</p>
<div class="block">
<div id="fig-spatial-table" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="fig/chap9/spatial_table.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Figure&nbsp;9: From https://rubd.github.io/Giotto_site/. (This figure is rapidly becoming out-dated. I would argue and say this figure is already nearly unrepresentative of the field.)</figcaption>
</figure>
</div>
</div>
<p>test</p>

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-figiel2024clinical" class="csl-entry" role="listitem">
Figiel, Sandy, Anthony Bates, David A Braun, Renu Eapen, Markus Eckstein, Brandon J Manley, Matthew I Milowsky, et al. 2024. <span>“Clinical Implications of Basic Research: Exploring the Transformative Potential of Spatial’omics in Uro-Oncology.”</span> <em>European Urology</em>.
</div>
<div id="ref-kim2023single" class="csl-entry" role="listitem">
Kim, Yoon-Seob, Jinyong Choi, and Sug Hyung Lee. 2023. <span>“Single-Cell and Spatial Sequencing Application in Pathology.”</span> <em>Journal of Pathology and Translational Medicine</em> 57 (1): 43–51.
</div>
<div id="ref-longo2021integrating" class="csl-entry" role="listitem">
Longo, Sophia K, Margaret G Guo, Andrew L Ji, and Paul A Khavari. 2021. <span>“Integrating Single-Cell and Spatial Transcriptomics to Elucidate Intercellular Tissue Dynamics.”</span> <em>Nature Reviews Genetics</em> 22 (10): 627–44.
</div>
<div id="ref-rademacher2024comparison" class="csl-entry" role="listitem">
Rademacher, Anne, Alik Huseynov, Michele Bortolomeazzi, Sina Jasmin Wille, Sabrina Schumacher, Pooja Sant, Denise Keitel, et al. 2024. <span>“Comparison of Spatial Transcriptomics Technologies Using Tumor Cryosections.”</span> <em>bioRxiv</em>, 2024–04.
</div>
<div id="ref-vanrobaeys2023spatial" class="csl-entry" role="listitem">
Vanrobaeys, Yann, Zeru J Peterson, Emily N Walsh, Snehajyoti Chatterjee, Li-Chun Lin, Lisa C Lyons, Thomas Nickl-Jockschat, and Ted Abel. 2023. <span>“Spatial Transcriptomics Reveals Unique Gene Expression Changes in Different Brain Regions After Sleep Deprivation.”</span> <em>Nature Communications</em> 14 (1): 7095.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>