---
title: "Single-cell epigenetics"
subtitle: "Transcription factor scoring, diagonal integration, gene regulatory networks"
format:
  html:
    toc: true
    toc-depth: 4
    number-sections: true
    theme: cosmo      # pick any Bootstrap theme you like
    css: styles.css   # provides the coloured ‘block’ boxes
execute:
  echo: false         # hide code by default
bibliography: references.bib   # optional .bib file for citations
---

## Primer on the genome, epigenetics, and enhancers

The genome is the complete set of DNA within an organism, encoding the instructions for life. While every cell in an organism typically contains the same genome, different cells exhibit distinct phenotypes and functions. This diversity arises not from changes in the underlying DNA sequence but from epigenetic regulation—heritable modifications that influence gene expression without altering the DNA itself. Epigenetics includes processes like DNA methylation, histone modification, and chromatin accessibility, all of which contribute to the dynamic regulation of gene activity in response to developmental cues and environmental signals.

Enhancers play a critical role in this regulatory landscape. These are DNA sequences that, while not coding for proteins themselves, can dramatically increase the transcription of target genes. Enhancers act by binding specific transcription factors, proteins that recognize and attach to DNA sequences to regulate gene expression. Some transcription factors require assistance from chaperone proteins, which ensure their proper folding and functionality, or pioneer proteins, which can access and open tightly packed chromatin to allow other factors to bind. This interplay highlights the complexity of the regulatory machinery that governs cellular function. See @fig-enhancer-promoter1 and @fig-enhancer-promoter2 to appreciate how complex this machinery is.

::: block
![From @ito2022factors.](fig/chap5/enhancer-promoter.png){#fig-enhancer-promoter1 width="100%" fig-align="center"}
:::

::: block
![From @claringbould2021enhancers.](fig/chap5/enhancer.png){#fig-enhancer-promoter2 width="100%" fig-align="center"}
:::

## The zoo of epigenetic modalities

Epigenetics encompasses a vast array of molecular mechanisms that regulate gene expression without altering the underlying DNA sequence. These mechanisms include modifications to DNA, RNA, chromatin, and the spatial organization of the genome, collectively forming a complex regulatory landscape. Below are some of the key modalities studied in epigenetics:

- **DNA Accessibility**: Techniques like ATAC-seq and DNase-seq measure how accessible DNA is to transcription factors and other regulatory proteins. Accessible regions often overlap with promoters and enhancers, providing critical insights into gene regulation. (This is what we'll focus on this chapter.)

- **DNA Methylation**[^1]: This modification, typically at cytosines in CpG dinucleotides, is a key epigenetic mark associated with gene silencing. Tools like bisulfite sequencing are used to map methylation patterns across the genome, revealing their roles in development and disease. See @fig-methylation1-1.

::: block
![(Left): From @gauba2021immunomodulation.](fig/chap5/methylation1.png){#fig-methylation1-1 width="100%" fig-align="center"}
:::

::: {#fig-methylation1-2}
![](fig/chap5/methylation2.png){width="100%" fig-align="center"}
(Right): From @bruggeman2008using.
:::

- **Hi-C and Genome Organization**[^2]: Hi-C measures chromatin interactions to reveal the three-dimensional structure of the genome. It uncovers features like topologically associating domains (TADs) and enhancer-promoter loops, which are crucial for understanding how spatial organization influences gene regulation. See @fig-hic1-1.

::: {#fig-hic1-1}
![](fig/chap5/hic1.png){width="100%" fig-align="center"}
(Left): From @ea2015contribution.
:::
::: {#fig-hic1-2}
![](fig/chap5/hic2.png){width="100%" fig-align="center"}
(Right): From @liu2021seeing.
:::

- **Histone Modifications**[^3]: Post-translational modifications, such as acetylation, methylation, and phosphorylation, occur on histone proteins and regulate chromatin structure. Techniques like ChIP-seq and the newer Cut & Tag method are used to map these modifications and their role in gene expression at the bulk level. See @fig-histone1.

::: {#fig-histone1}
![](fig/chap5/histone1.png){width="100%" fig-align="center"}
(Left): From <https://opened.cuny.edu/courseware/lesson/684/student/?section=2>.
:::
::: {#fig-histone2}
![](fig/chap5/histone2.png){width="100%" fig-align="center"}
(Right): From @steinbach2017pten. (Note: This is an example of the "canonical" functions of common histone modifications. This is by no means exhaustive and guaranteed to hold true for all biological systems.)
:::

Some of you might be interested: Personally, I think one of the fascinating concepts based on histone modifications is *bivalent chromatin* (essentially, chromatin that is wrapped in such a way that is simultaneously activated and silenced), see @fig-bivalent1-1. It's a particularly curious phenomenon that entire labs dedicate themselves to studying. See @blanco2020bivalent for an overview why this mechanism might be "beneficial."

::: {#fig-bivalent1-1}
![](fig/chap5/bivalent1.png){width="100%" fig-align="center"}
(Top): From @macrae2023regulation.
:::
::: {#fig-bivalent1-2}
![](fig/chap5/bivalent2.png){width="100%" fig-align="center"}
(Bottom): From @glancy2024bivalent.
:::

- **RNA Modifications (m6A and Pseudouridine)**: Modifications like N6-methyladenosine (m6A) and pseudouridine occur on RNA molecules and are involved in processes like splicing, translation, and mRNA decay. See @fig-pseudouridine for what pseudouridine is, i.e., "a rotation of the uridine molecule." These modifications add an epitranscriptomic layer to gene regulation.

::: {#fig-pseudouridine}
![](fig/chap5/pseudouridine.png){width="100%" fig-align="center"}
From @hamma2006pseudouridine about pseudouridine.
:::

- **Untranslated Regions (UTRs)**: Before talking about UTRs, it's probably good to review what an mRNA fragment "looks like" at the different stages of transcription and translation, see @fig-utr2. The 5′ and 3′ UTRs contain regulatory elements that influence mRNA stability, localization, and translation efficiency. The 3′ UTR, in particular, serves as a binding platform for RNA-binding proteins and microRNAs, providing an additional layer of post-transcriptional gene regulation.

::: {#fig-utr2}
![](fig/chap5/utr2.png){width="100%" fig-align="center"}
From @niazi2023rna. Notice the splicing (i.e., "removal" of the introns – see *Alternative Splicing* below) and that the protein-coding region is *not* the only part of the mRNA transcript – there are also the UTRs and poly-A tail. **NOTE**: The transcription start site (TSS) is *not* the same place as the promoter. The promoter is usually an un-transcribed region upstream of the TSS, while the TSS is where transcription actually starts.
:::
::: {#fig-utr1}
![](fig/chap5/utr1.png){width="100%" fig-align="center"}
(Top): From @mignone2002untranslated.
:::
::: {#fig-utr3}
![](fig/chap5/utr3.png){width="100%" fig-align="center"}
(Bottom): From <https://www.cambio.co.uk/9/257/applications/methods/in-vitro-transcription/mrna-therapeutics/>.
:::

- **Alternative Splicing**: This post-transcriptional process generates multiple mRNA isoforms from the same gene, expanding the proteomic diversity, see @fig-alternativesplicing1. **This is a combinatorial explosion of isoforms.** While not traditionally classified as epigenetic, splicing often intersects with chromatin modifications and RNA-binding proteins, blurring the lines between transcriptional and post-transcriptional regulation.

::: {#fig-alternativesplicing1}
![](fig/chap5/alternativesplicing1.png){width="100%" fig-align="center"}
From @chen2015alternative.
:::

- **Alternative Polyadenylation (APA)**: APA generates transcript isoforms with different 3′ ends, affecting mRNA stability, localization, and translation, see @fig-apa1. By selecting distinct cleavage sites, APA alters the *3′ UTR length* without affecting the poly-A tail itself, thereby modulating interactions with regulatory factors such as microRNAs and RNA-binding proteins.

::: {#fig-apa1}
![](fig/chap5/apa1.png){width="100%" fig-align="center"}
From @li2022population.
:::

Efforts like the ENCODE project (<https://pmc.ncbi.nlm.nih.gov/articles/PMC7061942/>) have systematically mapped these modalities across cell types and tissues, creating a comprehensive resource for understanding genome function. ENCODE has provided invaluable datasets on DNA accessibility, histone modifications, and RNA-binding proteins, enabling researchers to uncover how epigenetic and transcriptomic layers work together to drive cellular processes.

Finally, tools like MPRA (Massively Parallel Reporter Assays) are revolutionizing how we study enhancers and regulatory sequences. MPRA allows researchers to test thousands of DNA fragments for their regulatory activity, providing functional validation for epigenetic marks. This growing zoo of modalities continues to expand our understanding of how the genome is dynamically regulated in health and disease.

See @fig-epigenetic2-1 for some courageous figures that try to display multiple types of epigenetic modifications all at once. See @lim2024advances for a broad overview about the different types of technologies to sequence different omics and layers of epigenetics, and how they are getting computationally put together.

::: {#fig-epigenetic2-1}
![](fig/chap5/epigenetic2.png){width="100%" fig-align="center"}
(Left): From <https://www.sc-best-practices.org/chromatin_accessibility/introduction.html>.
:::
::: {#fig-epigenetic2-2}
![](fig/chap5/epigenetic3.png){width="100%" fig-align="center"}
(Right): From <https://www.whatisepigenetics.com/type-2-diabetes-mellitus-and-epigenetics>.
:::

[^1]: The technology to measure this at single-cell resolution is still being developed (see snmC-seq2 @liu2021dna), and is particularly interesting due to methylation's relation to cellular memory @kim2017dna and molecular clocks @hernando2019ageing; @trapp2021profiling; @gabbutt2022fluctuating.  

[^2]: See Droplet-HiC @chang2024droplet for an example of single-cell Hi-C.  

[^3]: See Paired-tag @zhu2021joint for an example of single-cell histone modification.

