<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<<<<<<< HEAD
<<<<<<< HEAD
<meta name="generator" content="quarto-1.7.32">
=======
<meta name="generator" content="quarto-1.5.57">
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<meta name="generator" content="quarto-1.5.57">
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<<<<<<< HEAD
<<<<<<< HEAD
<title>9&nbsp; Single-cell lineage tracing – scOmicNotes</title>
=======
<title>7&nbsp; Single-cell lineage tracing – scOmicNotes</title>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<title>7&nbsp; Single-cell lineage tracing – scOmicNotes</title>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
<<<<<<< HEAD
<<<<<<< HEAD
</style>
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<<<<<<< HEAD
<<<<<<< HEAD
<link href="./chapter9_spatial.html" rel="next">
<link href="./chapter7_crispr.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
=======
<link href="./references.html" rel="next">
<link href="./chapter5_epigenetics.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<link href="./references.html" rel="next">
<link href="./chapter5_epigenetics.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<<<<<<< HEAD
<<<<<<< HEAD
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-485d01fc63b59abcd3ee1bf1e8e2748d.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<<<<<<< HEAD
<<<<<<< HEAD
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634

<link rel="stylesheet" href="styles.css">
</head>

<<<<<<< HEAD
<<<<<<< HEAD
<body class="nav-sidebar floating quarto-light">
=======
<body class="nav-sidebar floating">
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<body class="nav-sidebar floating">
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
<<<<<<< HEAD
<<<<<<< HEAD
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./chapter8_lineage.html"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></a></li></ol></nav>
=======
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./chapter8_lineage.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></a></li></ol></nav>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./chapter8_lineage.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></a></li></ol></nav>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">scOmicNotes</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
<<<<<<< HEAD
<<<<<<< HEAD
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Preface-nello version</span></span></a>
=======
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Preface</span></span></a>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Preface</span></span></a>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2_sequencing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Single-cell sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3_rna.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Single-cell RNA-sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4_protein.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">‘Single-cell’ proteomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5_epigenetics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Single-cell epigenetics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
<<<<<<< HEAD
<<<<<<< HEAD
  <a href="./chapter6_dna.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">‘Single-cell’ DNA</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7_crispr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Single-cell CRISPR editting</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8_lineage.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9_spatial.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">‘Single-cell’ spatial transcriptomics</span></span></a>
=======
  <a href="./chapter8_lineage.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></span></a>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
  <a href="./chapter8_lineage.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></span></a>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
<<<<<<< HEAD
<<<<<<< HEAD
  <li><a href="#why-are-we-interested-in-learning-temporal-dynamics" id="toc-why-are-we-interested-in-learning-temporal-dynamics" class="nav-link active" data-scroll-target="#why-are-we-interested-in-learning-temporal-dynamics"><span class="header-section-number">9.1</span> Why are we interested in learning temporal dynamics?</a></li>
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
  <li><a href="#why-are-we-interested-in-learning-temporal-dynamics" id="toc-why-are-we-interested-in-learning-temporal-dynamics" class="nav-link active" data-scroll-target="#why-are-we-interested-in-learning-temporal-dynamics"><span class="header-section-number">7.1</span> Why are we interested in learning temporal dynamics?</a></li>
  <li><a href="#sec-natural_lineage" id="toc-sec-natural_lineage" class="nav-link" data-scroll-target="#sec-natural_lineage"><span class="header-section-number">7.2</span> Basis on how to infer history from cells naturally</a></li>
  <li><a href="#brief-aside-relation-to-snapshot-trajectory-inference" id="toc-brief-aside-relation-to-snapshot-trajectory-inference" class="nav-link" data-scroll-target="#brief-aside-relation-to-snapshot-trajectory-inference"><span class="header-section-number">7.3</span> Brief aside: Relation to “snapshot” trajectory inference</a></li>
  <li><a href="#reconstruction-phylogenetic-tree-from-sclt" id="toc-reconstruction-phylogenetic-tree-from-sclt" class="nav-link" data-scroll-target="#reconstruction-phylogenetic-tree-from-sclt"><span class="header-section-number">7.4</span> Reconstruction phylogenetic tree from scLT</a>
  <ul>
  <li><a href="#brief-review-number-of-rooted-trees" id="toc-brief-review-number-of-rooted-trees" class="nav-link" data-scroll-target="#brief-review-number-of-rooted-trees"><span class="header-section-number">7.4.1</span> Brief review: Number of rooted trees</a></li>
  <li><a href="#camin-sokal-maximum-parsimony-method" id="toc-camin-sokal-maximum-parsimony-method" class="nav-link" data-scroll-target="#camin-sokal-maximum-parsimony-method"><span class="header-section-number">7.4.2</span> Camin-Sokal maximum parsimony method</a></li>
  </ul></li>
  <li><a href="#lentiviral-lineage-tracing" id="toc-lentiviral-lineage-tracing" class="nav-link" data-scroll-target="#lentiviral-lineage-tracing"><span class="header-section-number">7.5</span> Lentiviral lineage tracing</a>
  <ul>
  <li><a href="#sec-lentiviral_lineage" id="toc-sec-lentiviral_lineage" class="nav-link" data-scroll-target="#sec-lentiviral_lineage"><span class="header-section-number">7.5.1</span> About the technology and application</a></li>
  <li><a href="#for-more-reading-on-this-technology" id="toc-for-more-reading-on-this-technology" class="nav-link" data-scroll-target="#for-more-reading-on-this-technology"><span class="header-section-number">7.5.2</span> For more reading on this technology</a></li>
  <li><a href="#cassiopeia-for-sclt-barcodes-illustrating-a-non-probabilistic-modeling-approach" id="toc-cassiopeia-for-sclt-barcodes-illustrating-a-non-probabilistic-modeling-approach" class="nav-link" data-scroll-target="#cassiopeia-for-sclt-barcodes-illustrating-a-non-probabilistic-modeling-approach"><span class="header-section-number">7.5.3</span> Cassiopeia, for scLT barcodes: Illustrating a non-probabilistic modeling approach</a></li>
  <li><a href="#some-implementation-notes" id="toc-some-implementation-notes" class="nav-link" data-scroll-target="#some-implementation-notes"><span class="header-section-number">7.5.4</span> Some implementation notes</a></li>
  <li><a href="#im-very-confused.-can-you-give-me-some-more-details" id="toc-im-very-confused.-can-you-give-me-some-more-details" class="nav-link" data-scroll-target="#im-very-confused.-can-you-give-me-some-more-details"><span class="header-section-number">7.5.5</span> I’m very confused. Can you give me some more details?</a></li>
  <li><a href="#a-brief-note-on-other-approaches" id="toc-a-brief-note-on-other-approaches" class="nav-link" data-scroll-target="#a-brief-note-on-other-approaches"><span class="header-section-number">7.5.6</span> A brief note on other approaches</a></li>
  </ul></li>
  <li><a href="#natural-lineage-barcodes-via-copy-number-alterations-cna" id="toc-natural-lineage-barcodes-via-copy-number-alterations-cna" class="nav-link" data-scroll-target="#natural-lineage-barcodes-via-copy-number-alterations-cna"><span class="header-section-number">7.6</span> Natural lineage barcodes via copy number alterations (CNA)</a>
  <ul>
  <li><a href="#about-the-technology-and-application" id="toc-about-the-technology-and-application" class="nav-link" data-scroll-target="#about-the-technology-and-application"><span class="header-section-number">7.6.1</span> About the technology and application</a></li>
  <li><a href="#sec-dna_phylogenetic" id="toc-sec-dna_phylogenetic" class="nav-link" data-scroll-target="#sec-dna_phylogenetic"><span class="header-section-number">7.6.2</span> COMPASS, for scDNA copy number: Illustrating a probabilistic approach</a></li>
  <li><a href="#a-brief-note-on-other-approaches-1" id="toc-a-brief-note-on-other-approaches-1" class="nav-link" data-scroll-target="#a-brief-note-on-other-approaches-1"><span class="header-section-number">7.6.3</span> A brief note on other approaches</a></li>
  </ul></li>
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<<<<<<< HEAD
<<<<<<< HEAD
<h1 class="title"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></h1>
=======
<h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></h1>
<p class="subtitle lead">[]</p>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></h1>
<p class="subtitle lead">[]</p>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<<<<<<< HEAD
<<<<<<< HEAD
<section id="why-are-we-interested-in-learning-temporal-dynamics" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="why-are-we-interested-in-learning-temporal-dynamics"><span class="header-section-number">9.1</span> Why are we interested in learning temporal dynamics?</h2>
<p>Studying how cells change over time provides critical insight into the sequence of events driving biological processes. By observing changes in cell populations across multiple time points, researchers can pinpoint when specific transitions or bifurcations occur. Such temporal information reveals which factors influence a cell’s fate and how quickly new traits emerge.</p>
<p>Moreover, understanding temporal dynamics can help us develop better interventions. If we can identify the earliest signs of disease or undesirable changes in cells, then targeted therapies can be designed to prevent or slow progression. Such strategies are especially powerful for complex, multi-stage diseases where later intervention might be less effective.</p>
<ul>
<li><strong>Cancer subclone evolution</strong>: Tumors consist of various subclones that compete and evolve. Observing which subclones become dominant over time helps illuminate how certain cells acquire and propagate new mutations, sometimes conferring resistance to treatments. See <a href="#fig-cancer_treatment1" class="quarto-xref">Figure&nbsp;<span>9.1</span></a> and <a href="#fig-cancer_treatment2" class="quarto-xref">Figure&nbsp;<span>9.2</span></a>.</li>
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
<section id="why-are-we-interested-in-learning-temporal-dynamics" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="why-are-we-interested-in-learning-temporal-dynamics"><span class="header-section-number">7.1</span> Why are we interested in learning temporal dynamics?</h2>
<p>Studying how cells change over time provides critical insight into the sequence of events driving biological processes. By observing changes in cell populations across multiple time points, researchers can pinpoint when specific transitions or bifurcations occur. Such temporal information reveals which factors influence a cell’s fate and how quickly new traits emerge.</p>
<p>Moreover, understanding temporal dynamics can help us develop better interventions. If we can identify the earliest signs of disease or undesirable changes in cells, then targeted therapies can be designed to prevent or slow progression. Such strategies are especially powerful for complex, multi-stage diseases where later intervention might be less effective.</p>
<ul>
<li><strong>Cancer subclone evolution</strong>: Tumors consist of various subclones that compete and evolve. Observing which subclones become dominant over time helps illuminate how certain cells acquire and propagate new mutations, sometimes conferring resistance to treatments. See <a href="#fig-cancer_treatment1" class="quarto-xref">Figure&nbsp;<span>7.1</span></a> and <a href="#fig-cancer_treatment2" class="quarto-xref">Figure&nbsp;<span>7.2</span></a>.</li>
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</ul>
<div class="block">
<div id="fig-cancer_treatment1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cancer_treatment1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/cancer_treatment1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cancer_treatment1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<<<<<<< HEAD
<<<<<<< HEAD
Figure&nbsp;9.1: (Top) From <span class="citation" data-cites="ashouri2023decoding">(<a href="references.html#ref-ashouri2023decoding" role="doc-biblioref"><strong>ashouri2023decoding?</strong></a>)</span>.
=======
Figure&nbsp;7.1: (Top) From <span class="citation" data-cites="ashouri2023decoding">Ashouri, Zhang, and Gaiti (<a href="references.html#ref-ashouri2023decoding" role="doc-biblioref">2023</a>)</span>.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
Figure&nbsp;7.1: (Top) From <span class="citation" data-cites="ashouri2023decoding">Ashouri, Zhang, and Gaiti (<a href="references.html#ref-ashouri2023decoding" role="doc-biblioref">2023</a>)</span>.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</figcaption>
</figure>
</div>
</div>
<div class="block">
<div id="fig-cancer_treatment2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cancer_treatment2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/cancer_treatment2.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cancer_treatment2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<<<<<<< HEAD
<<<<<<< HEAD
Figure&nbsp;9.2: (Bottom) From <span class="citation" data-cites="marine2020non">(<a href="references.html#ref-marine2020non" role="doc-biblioref"><strong>marine2020non?</strong></a>)</span>.
=======
Figure&nbsp;7.2: (Bottom) From <span class="citation" data-cites="marine2020non">Marine, Dawson, and Dawson (<a href="references.html#ref-marine2020non" role="doc-biblioref">2020</a>)</span>.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
Figure&nbsp;7.2: (Bottom) From <span class="citation" data-cites="marine2020non">Marine, Dawson, and Dawson (<a href="references.html#ref-marine2020non" role="doc-biblioref">2020</a>)</span>.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</figcaption>
</figure>
</div>
</div>
<ul>
<<<<<<< HEAD
<<<<<<< HEAD
<li><strong>Cancer metastasis</strong>: Cells that detach from the primary tumor site and successfully colonize new tissues undergo significant genetic and phenotypic changes. Understanding these changes in temporal sequence highlights the adaptations needed for invasion and survival in distant environments. See <a href="#fig-metastasis" class="quarto-xref">Figure&nbsp;<span>9.3</span></a>.</li>
=======
<li><strong>Cancer metastasis</strong>: Cells that detach from the primary tumor site and successfully colonize new tissues undergo significant genetic and phenotypic changes. Understanding these changes in temporal sequence highlights the adaptations needed for invasion and survival in distant environments. See <a href="#fig-metastasis" class="quarto-xref">Figure&nbsp;<span>7.3</span></a>.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<li><strong>Cancer metastasis</strong>: Cells that detach from the primary tumor site and successfully colonize new tissues undergo significant genetic and phenotypic changes. Understanding these changes in temporal sequence highlights the adaptations needed for invasion and survival in distant environments. See <a href="#fig-metastasis" class="quarto-xref">Figure&nbsp;<span>7.3</span></a>.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</ul>
<div class="block">
<div id="fig-metastasis" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-metastasis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/metastasis.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-metastasis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<<<<<<< HEAD
<<<<<<< HEAD
Figure&nbsp;9.3: From <span class="citation" data-cites="fu2023emerging">(<a href="references.html#ref-fu2023emerging" role="doc-biblioref"><strong>fu2023emerging?</strong></a>)</span>.
=======
Figure&nbsp;7.3: From <span class="citation" data-cites="fu2023emerging">(<a href="references.html#ref-fu2023emerging" role="doc-biblioref"><strong>fu2023emerging?</strong></a>)</span>.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
Figure&nbsp;7.3: From <span class="citation" data-cites="fu2023emerging">(<a href="references.html#ref-fu2023emerging" role="doc-biblioref"><strong>fu2023emerging?</strong></a>)</span>.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</figcaption>
</figure>
</div>
</div>
<ul>
<<<<<<< HEAD
<<<<<<< HEAD
<li><strong>Disease progression</strong>: Many diseases advance in stages, even beyond cancer, with cells accumulating subtle changes that eventually manifest as severe pathologies. Timing the acquisition of these changes reveals how early molecular events cascade into full-blown disease. See <a href="#fig-covid" class="quarto-xref">Figure&nbsp;<span>9.4</span></a> for an example of this investigated in COVID.</li>
=======
<li><strong>Disease progression</strong>: Many diseases advance in stages, even beyond cancer, with cells accumulating subtle changes that eventually manifest as severe pathologies. Timing the acquisition of these changes reveals how early molecular events cascade into full-blown disease. See <a href="#fig-covid" class="quarto-xref">Figure&nbsp;<span>7.4</span></a> for an example of this is investigated in COVID.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<li><strong>Disease progression</strong>: Many diseases advance in stages, even beyond cancer, with cells accumulating subtle changes that eventually manifest as severe pathologies. Timing the acquisition of these changes reveals how early molecular events cascade into full-blown disease. See <a href="#fig-covid" class="quarto-xref">Figure&nbsp;<span>7.4</span></a> for an example of this is investigated in COVID.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</ul>
<div class="block">
<div id="fig-covid" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-covid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/covid.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-covid-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<<<<<<< HEAD
<<<<<<< HEAD
Figure&nbsp;9.4: From <span class="citation" data-cites="stephenson2021single">(<a href="references.html#ref-stephenson2021single" role="doc-biblioref"><strong>stephenson2021single?</strong></a>)</span>. You can see that at different stages of COVID, there is a slightly different proportion of cell types that constitute the immune system.
=======
Figure&nbsp;7.4: From <span class="citation" data-cites="stephenson2021single">Stephenson et al. (<a href="references.html#ref-stephenson2021single" role="doc-biblioref">2021</a>)</span>. You can see that at different stages of COVID, there is a slightly different proportion of cell types that constitute the immune system.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
Figure&nbsp;7.4: From <span class="citation" data-cites="stephenson2021single">Stephenson et al. (<a href="references.html#ref-stephenson2021single" role="doc-biblioref">2021</a>)</span>. You can see that at different stages of COVID, there is a slightly different proportion of cell types that constitute the immune system.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</figcaption>
</figure>
</div>
</div>
<ul>
<<<<<<< HEAD
<<<<<<< HEAD
<li><strong>Embryonic/organ development and cell fate</strong>: During embryonic development, cells undergo a series of tightly regulated fate decisions that determine their final identity. These decisions are influenced by both intrinsic genetic programs and extrinsic signaling cues from the surrounding environment. Understanding the temporal dynamics of these transitions allows researchers to uncover the molecular mechanisms guiding differentiation and tissue formation. By tracking how cells commit to specific lineages, we gain insight into how organs form or how “cells make decisions”. See <a href="#fig-larry" class="quarto-xref">Figure&nbsp;<span>9.5</span></a>.</li>
=======
<li><strong>Embryonic/organ development and cell fate</strong>: During embryonic development, cells undergo a series of tightly regulated fate decisions that determine their final identity. These decisions are influenced by both intrinsic genetic programs and extrinsic signaling cues from the surrounding environment. Understanding the temporal dynamics of these transitions allows researchers to uncover the molecular mechanisms guiding differentiation and tissue formation. By tracking how cells commit to specific lineages, we gain insight into how organs form or how “cells make decisions”. See <a href="#fig-larry" class="quarto-xref">Figure&nbsp;<span>7.5</span></a>.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<li><strong>Embryonic/organ development and cell fate</strong>: During embryonic development, cells undergo a series of tightly regulated fate decisions that determine their final identity. These decisions are influenced by both intrinsic genetic programs and extrinsic signaling cues from the surrounding environment. Understanding the temporal dynamics of these transitions allows researchers to uncover the molecular mechanisms guiding differentiation and tissue formation. By tracking how cells commit to specific lineages, we gain insight into how organs form or how “cells make decisions”. See <a href="#fig-larry" class="quarto-xref">Figure&nbsp;<span>7.5</span></a>.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</ul>
<div class="block">
<div id="fig-larry" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-larry-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/larry.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-larry-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<<<<<<< HEAD
<<<<<<< HEAD
Figure&nbsp;9.5: From <span class="citation" data-cites="weinreb2020lineage">(<a href="references.html#ref-weinreb2020lineage" role="doc-biblioref"><strong>weinreb2020lineage?</strong></a>)</span>. This dataset is recorded longitudinally (i.e., cells were sequenced at Day 2, 4, and 6.) What’s shown here are cells known to be in different lineages, but certain lineages don’t differentiate by Day 6, differentiate into multiple cell types, or differentiate only into one cell type.
=======
Figure&nbsp;7.5: From <span class="citation" data-cites="weinreb2020lineage">Weinreb et al. (<a href="references.html#ref-weinreb2020lineage" role="doc-biblioref">2020</a>)</span>. This dataset is recorded longitudinally (i.e., cells were sequenced at Day 2, 4, and 6.) What’s shown here are cells known to be in different lineages, but certain lineages don’t differentiate by Day 6, differentiate into multiple cell types, or differentiate only into one cell type.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
Figure&nbsp;7.5: From <span class="citation" data-cites="weinreb2020lineage">Weinreb et al. (<a href="references.html#ref-weinreb2020lineage" role="doc-biblioref">2020</a>)</span>. This dataset is recorded longitudinally (i.e., cells were sequenced at Day 2, 4, and 6.) What’s shown here are cells known to be in different lineages, but certain lineages don’t differentiate by Day 6, differentiate into multiple cell types, or differentiate only into one cell type.
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</figcaption>
</figure>
</div>
</div>
<ul>
<<<<<<< HEAD
<<<<<<< HEAD
<li><strong>Stem cell research</strong>: Stem cells differentiate into specialized cell types following a tightly regulated timeline. Tracking these progressions uncovers the signals that guide each step and may inform regenerative medicine strategies for repairing damaged tissues. See <a href="#fig-ipsc" class="quarto-xref">Figure&nbsp;<span>9.6</span></a>.</li>
=======
<li><strong>Stem cell research</strong>: Stem cells differentiate into specialized cell types following a tightly regulated timeline. Tracking these progressions uncovers the signals that guide each step and may inform regenerative medicine strategies for repairing damaged tissues. See <a href="#fig-ipsc" class="quarto-xref">Figure&nbsp;<span>7.6</span></a>.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
<li><strong>Stem cell research</strong>: Stem cells differentiate into specialized cell types following a tightly regulated timeline. Tracking these progressions uncovers the signals that guide each step and may inform regenerative medicine strategies for repairing damaged tissues. See <a href="#fig-ipsc" class="quarto-xref">Figure&nbsp;<span>7.6</span></a>.</li>
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</ul>
<div class="block">
<div id="fig-ipsc" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ipsc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/ipsc.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ipsc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<<<<<<< HEAD
<<<<<<< HEAD
Figure&nbsp;9.6: From <a href="https://stemcellthailand.org/induced-pluripotent-stem-cells-ips-ipscs-hipscs/" class="uri">https://stemcellthailand.org/induced-pluripotent-stem-cells-ips-ipscs-hipscs/</a>.
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
Figure&nbsp;7.6: From <a href="https://stemcellthailand.org/induced-pluripotent-stem-cells-ips-ipscs-hipscs/" class="uri">https://stemcellthailand.org/induced-pluripotent-stem-cells-ips-ipscs-hipscs/</a>.
</figcaption>
</figure>
</div>
</div>
</section>
<section id="sec-natural_lineage" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="sec-natural_lineage"><span class="header-section-number">7.2</span> Basis on how to infer history from cells naturally</h2>
<p>Certain features in a cell’s genetic material serve as natural markers for tracing lineage. These features arise spontaneously or accumulate over time, and researchers can compare them across related cells to reconstruct branching histories.</p>
<ul>
<li><p><strong>Mitochondrial DNA (somatic mutations)</strong>:<br>
Mitochondrial genomes can pick up small mutations as cells divide. (Note! There are a lot of types of other DNA in your cells beyond the typical DNA we think of in the 23 pairs of chromosomes. Your mitochondria has its own DNA.) By measuring these mutations, one can infer which cells share a common ancestor and roughly estimate how many divisions have occurred. However, because the mutation rate is not always high, it can be difficult to achieve fine-grained resolution.</p></li>
<li><p><strong>DNA copy number</strong>:<br>
Gains or losses in chromosome segments, known as copy number variations, can also serve as lineage markers. Though useful for distinguishing large-scale genomic shifts, such variations do not necessarily arise in a predictable manner. This makes it challenging to reproducibly track precise cell-to-cell histories.<br>
For examples of bulk-DNA phylogenetic tree, see . This is a massive field, so one citation is not doing this field justice. We’ll see this more in <a href="#sec-dna_phylogenetic">Section 8.3</a>.</p></li>
<li><p><strong>DNA methylation</strong>:<br>
DNA methylation serves as another naturally occurring cellular barcode. Over time, cells acquire unique methylation patterns that reflect their lineage history. These epigenetic marks regulate gene expression and can be stably inherited through cell divisions. Because different tissues and cell types exhibit distinct methylation landscapes, analyzing these patterns can provide a historical record of a cell’s developmental path. However, while DNA methylation is useful for lineage tracing, its resolution is limited by the variability in methylation turnover rates and the difficulty of distinguishing cell-to-cell differences in a reproducible manner.</p></li>
</ul>
<p>These naturally occurring barcodes, while informative, can be limiting in resolution. They may not mutate or appear frequently enough to trace detailed branching events. Additionally, reproducing the exact timing of such genetic changes in the wet lab can be difficult, making it hard to pinpoint when and how they impacted lineage formation.</p>
</section>
<section id="brief-aside-relation-to-snapshot-trajectory-inference" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="brief-aside-relation-to-snapshot-trajectory-inference"><span class="header-section-number">7.3</span> Brief aside: Relation to “snapshot” trajectory inference</h2>
<p>Snapshot-based methods, such as Monocle or Slingshot, rely on measuring gene expression states at a single time point (or a limited set of time points). They use computational models to infer the probable order and branching of cell fates based on these static profiles. While powerful, these methods rely on assumptions about the smoothness of transitions and may not capture rare events or subtle branching.</p>
<p>Lineage tracing complements snapshot-based methods by providing a direct record of which cells arose from the same ancestor. By confirming or refining the trajectories suggested by computational approaches, lineage tracing can validate inferred timelines, reveal overlooked branches, and identify rare subpopulations that might be missed in snapshot data alone. This integration leads to a more complete picture of cell fate decisions and improves our ability to predict and potentially manipulate cellular transitions.</p>
</section>
<section id="reconstruction-phylogenetic-tree-from-sclt" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="reconstruction-phylogenetic-tree-from-sclt"><span class="header-section-number">7.4</span> Reconstruction phylogenetic tree from scLT</h2>
<p>The goal in this section is to provide a high-level overview of how we can reconstruct phylogenetic trees from single-cell lineage tracing (scLT) dataset, if we are using mutating barcodes. Remember, in single-cell sequencing experiments, we never get to see anything “directly” aside from sequences (of RNA, DNA, etc.) So we need to “estimate” the phylogenetic tree itself using the barcode edits. See <a href="#fig-phylogenetic-goal1" class="quarto-xref">Figure&nbsp;<span>7.7</span></a> for a conceptual illustration of this problem.</p>
<div class="block">
<div id="fig-phylogenetic-goal1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-phylogenetic-goal1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/phylogenetic_goal1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-phylogenetic-goal1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.7: From
</figcaption>
</figure>
</div>
</div>
<p>Conceptually, this is our goal in a cartoon setting. (Technically, most lineage-tracing methods based on lentiviral barcodes return a <em>claudogram</em> (i.e., the edge lengths do not carry “time-information.” This is in contrast with a <em>phylogram</em>, where edge denote “amount of elapsed time since the branching event”). See <span class="quarto-unresolved-ref">?fig-tree1</span> and <span class="quarto-unresolved-ref">?fig-tree2</span>.</p>
<div class="block">
<p><img src="fig/chap8/tree.png" id="fig-tree1" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From https://microbenotes.com/phylogenetic-tree/"> <img src="fig/chap8/cladogram.png" id="fig-tree2" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From https://microbenotes.com/phylogenetic-tree/"></p>
</div>
<section id="brief-review-number-of-rooted-trees" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="brief-review-number-of-rooted-trees"><span class="header-section-number">7.4.1</span> Brief review: Number of rooted trees</h3>
<p>Given <span class="math inline">\(n\)</span> nodes, how many possible phylogenetic trees are possible? Couldn’t we just enumerate all possible trees with <span class="math inline">\(n\)</span> leaves, and determine which one is the “best one”? Think of this as a “caveman” optimization approach. All I need is a function that evaluates the “data agreement” score of a tree. Then, given single-cell data (either of scLT or copy numbers) of <span class="math inline">\(n\)</span> cells, I can write down all possible trees (agnostic of data), and then pick the tree with the score that best agrees with data. What’s wrong with this approach?</p>
<p>The issue is that there is an <em>explosion</em> of possible trees. Let’s establish some ground rules. We will only talk about fully connected, binary, rooted trees where the <span class="math inline">\(n\)</span> nodes are the leaves.</p>
<ul>
<li><p><strong><span class="math inline">\(n\)</span> leaves</strong>: The <span class="math inline">\(n\)</span> cells we observed are set to be leaves (sometimes called terminal nodes). (After all, we sequence all the cells at once, so we don’t actually observe the “parent” cell that divided to yield the two daughter cells we sequenced. Remember, over time, cells undergo mitosis to split into 2 cells.)</p></li>
<li><p><strong>Fully connected and directed</strong>: All <span class="math inline">\(n\)</span> leaves are connected by one graph. These <span class="math inline">\(n\)</span> nodes are connected by <span class="math inline">\(n-1\)</span> intermediate (unobserved) nodes.</p></li>
<li><p><strong>Binary and rooted</strong>: Every one of the <span class="math inline">\(n-1\)</span> intermediate nodes has 2 children nodes. These children nodes could be one of the <span class="math inline">\(n\)</span> leaves or one of the other intermediate nodes. Furthermore, every node has a parent <em>except</em> one special node called the “root node.” You can think of the root node as the common (unobserved) ancestor that spawned all the <span class="math inline">\(n\)</span> cells.</p></li>
</ul>
<div class="block">
<div id="fig-unrooted-rooted" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-unrooted-rooted-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/unrooted-rooted.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-unrooted-rooted-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.8: From <a href="https://microbenotes.com/phylogenetic-tree/" class="uri">https://microbenotes.com/phylogenetic-tree/</a>
</figcaption>
</figure>
</div>
</div>
<p>With these ground rules in place, we can prove the following result.</p>
<div class="block">
<p>Given <span class="math inline">\(n\)</span> named leaves, there are <span class="math inline">\((2n - 3)!!\)</span> (where the <span class="math inline">\(!!\)</span> denotes a double factorial<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>) possible rooted binary trees.</p>
</div>
<p>Here is an informal sketch of the proof.</p>
<div class="block2">
<p>We prove that the number of fully resolved (binary) rooted phylogenetic trees on <span class="math inline">\(n\)</span> labeled taxa (leaves) is given by</p>
<p><span id="eq-double-factorial"><span class="math display">\[
T_n = (2n - 3)!! = 1 \times 3 \times 5 \times \cdots \times (2n - 3),
\tag{7.1}\]</span></span></p>
<p>via induction.</p>
<p><strong>Base Case (<span class="math inline">\(n=2\)</span>)</strong>:<br>
For <span class="math inline">\(n=2\)</span>, there is exactly one fully resolved rooted tree, consisting of a root node with two children (the two labeled leaves). Hence,</p>
<p><span class="math display">\[
T_2 = 1.
\]</span></p>
<p><strong>Inductive Step</strong>:<br>
Assume that for <span class="math inline">\(n-1\)</span> leaves, the number of fully resolved rooted trees is given by <span class="math inline">\(T_{n-1}\)</span>. We show that the number of such trees for <span class="math inline">\(n\)</span> leaves satisfies the recurrence relation</p>
<p><span class="math display">\[
T_n = (2n - 3) \times T_{n-1}.
\]</span></p>
<p>To form an <span class="math inline">\(n\)</span>-leaf tree from a <span class="math inline">\((n-1)\)</span>-leaf tree, we consider a particular <span class="math inline">\((n-1)\)</span>-leaf tree and “split” one node. (By splitting, think of a “cherry” — a node with 2 children. We will take a node, set that to be a new internal node where one of its children is the original sub-tree of the node we picked, and the other child is the new leaf node for the <span class="math inline">\(n\)</span>-leaf tree.)</p>
<p>Hence, we need to count how many possible nodes are available to split. In a fully resolved tree with <span class="math inline">\(n-1\)</span> leaves, we have <span class="math inline">\(n-2\)</span> internal nodes (i.e., “unobserved cells”), one of which is the root, and therefore, we have a total of <span class="math inline">\((n-1)+(n-2) = 2n - 3\)</span> nodes.</p>
<p>Therefore, there are <span class="math inline">\((2n - 3)\)</span> such choices how to “split” a <span class="math inline">\((n-1)\)</span>-leaf tree.</p>
<p>Since there are <span class="math inline">\(T_{n-1}\)</span> possible <span class="math inline">\((n-1)\)</span>-leaf trees, (of which we have <span class="math inline">\((2n - 3)\)</span> choices on how to form a <span class="math inline">\(n\)</span>-leaf tree<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>), we have the relation,</p>
<p><span class="math display">\[
T_n = (2n - 3) \times T_{n-1}.
\]</span></p>
</div>
<p>For example, this means</p>
<p><span class="math display">\[
\begin{aligned}
T_3 &amp;= (2 \times 3 - 3) \times \underbrace{T_2}_{=1} = 3, \\
T_4 &amp;= (2 \times 4 - 3) \times \underbrace{T_3}_{=3} = (2 \times 4 - 3) \times (2 \times 3 - 3) \times 1 = 15.
\end{aligned}
\]</span></p>
<p>See <a href="#fig-tree-enumerate" class="quarto-xref">Figure&nbsp;<span>7.9</span></a> for the trees enumerated.<br>
Using the base case <span class="math inline">\(T_2 = 1\)</span>, repeated application of this recurrence gives</p>
<p><span class="math display">\[
T_n = (2n - 3)!!.
\]</span></p>
<p>:::</p>
<div class="block">
<div id="fig-tree-enumerate" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-tree-enumerate-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/tree-enumerate.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-tree-enumerate-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.9: From <a href="https://www.megasoftware.net/mega1_manual/Phylogeny.html" class="uri">https://www.megasoftware.net/mega1_manual/Phylogeny.html</a>
</figcaption>
</figure>
</div>
</div>
<p>See (and related papers) for more context.<br>
So… why is this result important? Observe that <span class="math inline">\(n!!\)</span> is <em>super exponential</em> (i.e., it grows faster than an exponential<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>). So if you have <span class="math inline">\(n=1000\)</span> cells, the number of possible trees you’ll try to enumerate is on the order of <span class="math inline">\(1000^{1000} = 10^{3000}\)</span>, which is way too many. See <a href="#fig-num-trees" class="quarto-xref">Figure&nbsp;<span>7.10</span></a>.</p>
<p>Therefore, when constructing phylogenetic trees, we need to be mathematically clever (probably with the help of some heuristic). Can we use some heuristic to drastically reduce the number of trees we’re going to even consider, and we can use our sequencing data of these <span class="math inline">\(n\)</span> cells to help us in this endeavor?</p>
<div class="block">
<div id="fig-num-trees" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-num-trees-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/num_trees.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-num-trees-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.10: From <a href="https://calla.rnet.missouri.edu/cheng_courses/slides9.pdf" class="uri">https://calla.rnet.missouri.edu/cheng_courses/slides9.pdf</a>
</figcaption>
</figure>
</div>
</div>
</section>
<section id="camin-sokal-maximum-parsimony-method" class="level3" data-number="7.4.2">
<h3 data-number="7.4.2" class="anchored" data-anchor-id="camin-sokal-maximum-parsimony-method"><span class="header-section-number">7.4.2</span> Camin-Sokal maximum parsimony method</h3>
<p>The Camin-Sokal maximum parsimony method is a phylogenetic reconstruction technique that aims to infer evolutionary relationships by minimizing the number of changes required to explain observed character states. (You’ll see this come up a <em>lot</em> when reading papers that reconstruct the phylogenetic tree from scLT data with mutating barcodes. You can think of this as the “baseline method” that every sc-phylogenetic method compares itself to.)</p>
<p>This method assumes that character states can only evolve in one direction, meaning once a trait is gained, it cannot be lost. Given a set of taxa and their character states, the algorithm constructs a tree that explains these states with the fewest evolutionary steps. While this method is computationally efficient for small datasets, it may oversimplify biological reality by enforcing irreversible character changes, limiting its applicability in cases where traits can reappear or revert.</p>
</section>
</section>
<section id="lentiviral-lineage-tracing" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="lentiviral-lineage-tracing"><span class="header-section-number">7.5</span> Lentiviral lineage tracing</h2>
<section id="sec-lentiviral_lineage" class="level3" data-number="7.5.1">
<h3 data-number="7.5.1" class="anchored" data-anchor-id="sec-lentiviral_lineage"><span class="header-section-number">7.5.1</span> About the technology and application</h3>
<p>Lentiviral approaches offer a more controlled way to label and track cell lineages. By inserting a known genetic barcode into cells, scientists can reliably follow how these labeled cells expand and diversify over time. See <span class="quarto-unresolved-ref">?fig-barcode1</span>, <span class="quarto-unresolved-ref">?fig-barcode2</span>, and <span class="quarto-unresolved-ref">?fig-barcode3</span>.</p>
<ul>
<li><p><strong>In static lineage barcoding</strong>, a lentivirus delivers a stable transgene (for example, GFP or RFP) containing a unique sequence in its untranslated region (UTR) to be integrated into the cell’s DNA. All progeny of an infected cell inherit this same sequence, making it straightforward to group related cells.</p>
<p>(This won’t be the focus of the following sections. In static lineage barcoding, there is no phylogenetic tree. All the cell lineage information is relative to when the lentivirus was introduced to your cells.)</p></li>
<li><p><strong>In mutating barcoding</strong> (or also called evolving barcode), a guide RNA (via CRISPR-Cas9) can introduce variations sometimes into a transgene reporter (a gene artificially integrated into the cell’s DNA) at various target sites.<br>
This creates barcodes as “genetic scars” in descendant cells. This dynamic approach can provide a richer record of how cells diverge and acquire distinct identities.</p>
<p>(There’s a lot more complexity to this. Here are some concerns that are a bit easier to explain:</p>
<ol type="1">
<li>You need make sure the Cas9 protein is continually being expressed over the course of the experiment.<br>
</li>
<li>You need to make sure you have enough diversity in the genetic scars that it’s unlikely two unrelated cells happened to acquire the same set of scars.<br>
</li>
<li>These CRISPR double-stranded breaks can lead to sometimes large deletion events. If there’s not a lot of different locations the transgene was integrated into the genome, or if the target sites are too close, then the multiple target sites might be deleted at once<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>.)</li>
</ol></li>
</ul>
<div class="block">
<p><img src="fig/chap8/larry-barcode.png" id="fig-barcode1" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From , an example of static barcode."> <img src="fig/chap8/cassiopeia_evolving.png" id="fig-barcode2" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From , an example of evolving barcode."> <img src="fig/chap8/lineage-tech1.png" id="fig-barcode3" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From ."></p>
</div>
<p>See for fantastic overviews on both types of technologies.</p>
<p>What can we do with this? As shown in <a href="#fig-lineage-application" class="quarto-xref">Figure&nbsp;<span>7.11</span></a>, in the cancer setting, we can use lineage barcoding to directly observe many qualities of cancer cells. See <a href="#fig-lineage-tech2" class="quarto-xref">Figure&nbsp;<span>7.12</span></a> for a broader overview of different technologies and what biological insights they can reveal (based on what kind of information we can reconstruct from the barcode).</p>
<div class="block">
<div id="fig-lineage-application" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lineage-application-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/lineage-application.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lineage-application-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.11: From
</figcaption>
</figure>
</div>
</div>
<div class="block">
<div id="fig-lineage-tech2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lineage-tech2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/lineage-tech2.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lineage-tech2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.12: From
</figcaption>
</figure>
</div>
</div>
</section>
<section id="for-more-reading-on-this-technology" class="level3" data-number="7.5.2">
<h3 data-number="7.5.2" class="anchored" data-anchor-id="for-more-reading-on-this-technology"><span class="header-section-number">7.5.2</span> For more reading on this technology</h3>
<p>See , all of which are reviews that document the various ways labs have designed systems to record cellular history.</p>
</section>
<section id="cassiopeia-for-sclt-barcodes-illustrating-a-non-probabilistic-modeling-approach" class="level3" data-number="7.5.3">
<h3 data-number="7.5.3" class="anchored" data-anchor-id="cassiopeia-for-sclt-barcodes-illustrating-a-non-probabilistic-modeling-approach"><span class="header-section-number">7.5.3</span> Cassiopeia, for scLT barcodes: Illustrating a non-probabilistic modeling approach</h3>
<p>In many CRISPR-lineage systems, each target site transitions exactly once from unedited to some specific indel state, and cannot revert (i.e., irreversible). This constraint helps reduce the multi-state problem to something akin to a binary “perfect phylogeny” assumption. However, real data can deviate from a true perfect phylogeny due to parallel evolution of the same indel or missing data.</p>
<div class="block3">
<p><strong>Input/Output.</strong> The input to Cassiopeia is the <span class="math inline">\(\{\texttt{NA}, 0, 1, 2, \ldots\} \in \mathbb{R}^{n \times m}\)</span> matrix of <span class="math inline">\(n\)</span> cells and <span class="math inline">\(m\)</span> possible target sites (i.e., Cas9-induced heritable insertions or deletions happened, also called indels). This is sometimes called the “character matrix”. Here, <code>NA</code> refers to “missing” (sometimes encoded as <span class="math inline">\(-1\)</span> instead). <code>0</code> refers to the baseline sequence (i.e., unedited). Any positive integer (<span class="math inline">\(1, 2, \ldots\)</span>) denotes a categorical variable, that is, the type of indel mutation that happened at this site.</p>
<p>The output is a phylogenetic tree that is rooted binary where each of the <span class="math inline">\(n\)</span> sequenced cells is a leaf, but this tree does not contain any temporal information about how “far in the past” the cell divisions happened (i.e., it’s a cladogram, not a phylogram).</p>
<p>Notice that no scRNA-seq (gene expression) matrix is needed. All the sequencing that’s needed was to determine the genetic modifications at the target sites for these <span class="math inline">\(n\)</span> cells.</p>
</div>
<p>Cassiopeia offers a couple different ways to construct its trees. We highlight two of these options below.</p>
<ul>
<li><p><strong>Cassiopeia-ILP (Steiner Tree Formulation)</strong>:<br>
First, Cassiopeia-ILP constructs a directed acyclic “potential graph” whose vertices represent all plausible ancestral states (including the <span class="math inline">\(n\)</span> observed leaves), and edges represent possible single-step mutations. The algorithm repeatedly merges pairs of nodes with a shared ancestor if they differ by a valid mutation, generating internal vertices. A user-defined parameter controls how large this potential graph may grow (to avoid excessive computational cost).</p>
<p>Once the potential graph <span class="math inline">\(G\)</span> is built, the algorithm solves the <em>Steiner tree problem</em> on <span class="math inline">\(G\)</span> with a flow-based integer linear programming (ILP). The ILP ensures each observed cell is reached from the root (the unmutated state) by a minimal set of edges, minimizing the total number of mutation events. This method is exact but can be computationally slow for large <span class="math inline">\(n\)</span>.</p>
<div class="block2">
<p>The Steiner tree problem is a combinatorial optimization problem that seeks the shortest possible network connecting a given set of points (terminals) while allowing additional intermediate points (Steiner nodes) to reduce the total length. In the context of evolutionary biology and phylogenetics, the Steiner tree approach can be used to infer ancestral relationships by identifying the most parsimonious connections between observed species. The problem is NP-hard, meaning exact solutions are computationally infeasible for large datasets. However, various heuristic and approximation algorithms have been developed to efficiently construct near-optimal Steiner trees in practical applications.</p>
<p>This is different from computing the minimal spanning tree because we don’t need the Steiner tree to include every node in <span class="math inline">\(G\)</span>. We just need it to contain the <span class="math inline">\(n\)</span> leaves (i.e., the cells we actually observed). There might be hypothetical intermediate nodes (i.e., ancestral cells) in <span class="math inline">\(G\)</span> that are not deemed necessary to be included in the output Steiner tree. See <span class="quarto-unresolved-ref">?fig-cassiopeia1</span> and <span class="quarto-unresolved-ref">?fig-cassiopeia2</span>.</p>
</div>
<p>(In short, you can imagine the first step of forming the potential graph <span class="math inline">\(G\)</span> is to limit the space of possible phylogenetic trees — the returned tree must be a tree formed by the edges in <span class="math inline">\(G\)</span>. Then, in the second step, we formulate an ILP reminiscent of the Steiner tree problem to find the most “appropriate” tree.)</p></li>
<li><p><strong>Cassiopeia-Greedy</strong>:<br>
This algorithm builds the tree top-down, repeatedly choosing the most “frequent” or highest-priority mutation (unmutated <span class="math inline">\(\to\)</span> indel) to partition the cells into two subsets: those that carry that mutation vs.&nbsp;those that do not (i.e., it’s “greedy”).</p>
<p>Because CRISPR edits are irreversible, each multi-state site behaves effectively like a one-time event from an uncut state. Thus, one can treat each new indel as a potential splitting event. Cells with missing values at the chosen splitting site are assigned to whichever subset their non-missing edits best match. This step recurses until subsets become single cells or until all characters are used.</p>
<p>Greedy is fast and scales to tens of thousands of cells, but may produce suboptimal solutions if extensive parallel evolution exists.</p></li>
</ul>
<div class="block">
<p><img src="fig/chap8/cassiopeia1.png" id="fig-cassiopeia1" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From "> <img src="fig/chap8/cassiopeia2.png" id="fig-cassiopeia2" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From "></p>
</div>
</section>
<section id="some-implementation-notes" class="level3" data-number="7.5.4">
<h3 data-number="7.5.4" class="anchored" data-anchor-id="some-implementation-notes"><span class="header-section-number">7.5.4</span> Some implementation notes</h3>
<ul>
<li><em>Priors on Indel Frequencies.</em> Optionally, Cassiopeia can leverage empirical estimates of each indel’s probability to prioritize rare-but-frequent states (mitigating parallel evolution).</li>
<li><em>Missing Data.</em> Both ILP and greedy approaches accommodate missing data by enumerating or heuristically assigning cells to the most compatible subset.</li>
</ul>
</section>
<section id="im-very-confused.-can-you-give-me-some-more-details" class="level3" data-number="7.5.5">
<h3 data-number="7.5.5" class="anchored" data-anchor-id="im-very-confused.-can-you-give-me-some-more-details"><span class="header-section-number">7.5.5</span> I’m very confused. Can you give me some more details?</h3>
<p>There’s a lot of moving components since the lineage barcode data for mutating barcodes is quite unlike anything else we’ve seen in this course so far. Let me go through an example from <a href="https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/reconstruct.html" class="uri">https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/reconstruct.html</a> (the Cassiopeia tutorial itself) to tell you what the “input data” for Cassiopeia is. (Once you understand the input data and how it was preprocessed, it’ll become clear what the output is.)</p>
<p>In a typical mutating lineage barcoding experiment, you get a bunch of gDNA or gRNA reads about the sequences of the barcodes that integrated into each cell. Then, you run bioinformatics code to align these sequences to a known reference genome to determine what happened at each target site of a cell’s barcode (i.e., did a mutation happen, if so, what was the insertion or deletion event, etc.). This is described in <a href="https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/preprocess.html" class="uri">https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/preprocess.html</a>. The output of this is an allele table that might look like <a href="#fig-cassiopeia-bc1" class="quarto-xref">Figure&nbsp;<span>7.13</span></a>.</p>
<div class="block">
<div id="fig-cassiopeia-bc1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cassiopeia-bc1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/cassiopeia_bc1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cassiopeia-bc1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.13: From <a href="https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/reconstruct.html" class="uri">https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/reconstruct.html</a>. We are showing 10 different integrated barcodes <code>intBC</code> (each with three target sites, <code>r1</code> to <code>r3</code>) for the cell with <code>RW.TTTGCGCCAAGAAAGG</code>.<br>
So you should expect every unique cell (i.e., unique <code>cellBC</code>) to have one or more integrated barcodes (for example, 10, as shown here). Furthermore, you hope that many different cells have the same integrated barcode, such as <code>CTAGGGCATCTACG</code>. (This is because you’re going to compare what happened at different target sites for cells that have the same integrated barcode to determine how these cells are related.)<br>
For instance, in integrated barcode <code>CTAGGGCATCTACG</code>, we see at the first target site, <code>CCGAA[None]AAATG</code>, the <code>None</code> means there was no insertion or deletion event found at this site (relative to all the other sequences at the <code>r1</code> region of this integrated barcode in the other cells).<br>
In contrast, at the third target site <code>r3</code>, we see <code>ATTCG[221:1D]GGAGG</code>. This means at relative position 221 in this target site, there was a deletion that resulted in 1 base pair being deleted. (You can see in this screenshot that some of the deletion events can go up to 80+ base pairs, such as <code>TGGCG[86:81D]CTCTG</code>.) This large deletion event is okay (as long as it didn’t result in the deletion of a different target site) if the sequence alignment step could still determine that the flanking sequences to the deletion (in this case, <code>TGGCG</code> to the left and <code>CTCTG</code> to the right) still aligned to the reasonable positions to the target site for this barcode. Remember, <em>you</em> know what the full sequences of all these integrated barcodes are because these are all sequences you designed to be integrated into the cell’s DNA. See <code>PCT48.ref.fasta</code> in <a href="https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/preprocess.html" class="uri">https://cassiopeia-lineage.readthedocs.io/en/latest/notebooks/preprocess.html</a>.
</figcaption>
</figure>
</div>
</div>
<p>This character matrix is the input to Cassiopeia — it now knows the type of mutation at each of the target sites for every cell, and then it’s going to try to estimate a phylogenetic tree from this.</p>
<p>This character matrix is the input to Cassiopeia — it now knows the type of mutation at each of the target sites for every cell, and then it’s going to try to estimate a phylogenetic tree from this.</p>
</section>
<section id="a-brief-note-on-other-approaches" class="level3" data-number="7.5.6">
<h3 data-number="7.5.6" class="anchored" data-anchor-id="a-brief-note-on-other-approaches"><span class="header-section-number">7.5.6</span> A brief note on other approaches</h3>
<p>Cassiopeia doesn’t directly tell you some crucial pieces of information that you might want to know:<br>
1) how far in the past was the division (i.e., the “branch length”), and<br>
2) how fast are mutations happening (i.e., the “mutation rate”)?</p>
<p>For these, you could invoke more involved methods such as GAPML or ICE-FASE . On a different line of thinking, methods like LinTIMaT rely on using the lineage barcoding data alongside the scRNA-seq data (which tells you how similar two cells are based on either their annotated cell-types or expressed pathways) to construct phylogenetic trees.</p>
</section>
</section>
<section id="natural-lineage-barcodes-via-copy-number-alterations-cna" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="natural-lineage-barcodes-via-copy-number-alterations-cna"><span class="header-section-number">7.6</span> Natural lineage barcodes via copy number alterations (CNA)</h2>
<section id="about-the-technology-and-application" class="level3" data-number="7.6.1">
<h3 data-number="7.6.1" class="anchored" data-anchor-id="about-the-technology-and-application"><span class="header-section-number">7.6.1</span> About the technology and application</h3>
<p>One of the goals of why studying copy number variation in <a href="#sec-dna_structural_variation">Section 8.3</a> was to understand cellular dynamics. So… how is this done?<br>
Copy number alterations (CNAs) are a hallmark of many cancers and arise due to genomic instability within tumor cells (refer back to <span class="quarto-unresolved-ref">?fig-copy-number-cancer</span>). Because cancer is often driven by such genomic instabilities, CNAs serve as a natural “barcode” of lineage. By tracking which genes or regions of chromosomes have been duplicated or lost over time, researchers can infer how subclones evolve and expand. These aberrations not only reveal how aggressive a tumor might be, but can also hint at its capacity for metastasis. Indeed, it is quite common to see particular copy number events that are associated with advanced disease states or resistance to therapy.</p>
<p>Another key advantage of using CNAs in cancer research is the ability to reconstruct phylogenetic trees of tumor evolution at single-cell resolution. Traditional bulk sequencing can obscure the heterogeneity of a tumor by averaging out signals across thousands or millions of cells, making it difficult to identify rare but important subpopulations. By examining individual cells, we can trace how specific lineages emerged, which mutations or copy number changes appeared first, and how these have proliferated. This level of detail helps us pinpoint the subclones that might be the most aggressive or metastatic, and may reveal new therapeutic targets based on the genetic changes that drove each lineage.</p>
<p>Furthermore, once a phylogenetic tree is built from copy number data, the specific chromosomal regions that underwent duplication or deletion can yield insight into the biology of the tumor. Some genomic loci harbor oncogenes or tumor suppressors, so gains or losses in these regions can be especially detrimental—or beneficial, from the tumor’s perspective. Such events might correlate with faster growth, immune evasion, or other clinically significant traits. By focusing on these recurrently altered regions, researchers and clinicians can identify potential biomarkers for prognosis or drug targets for personalized therapy. See <a href="#fig-dna-phylogenetic" class="quarto-xref">Figure&nbsp;<span>7.14</span></a> as an example of what this workflow looks like.</p>
<div class="block">
<div id="fig-dna-phylogenetic" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-dna-phylogenetic-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/dna_phylogenetic.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-dna-phylogenetic-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.14: From . Imagine we have longitudinal samples (bottom left) of a person. Those tissue samples get single-cell sequenced. Then a variant calling happens, and then we want to learn a clone tree (right middle), which is the focus for the type of tree we’re discussing in these notes.
</figcaption>
</figure>
</div>
</div>
<p>These output phylogenetic trees might look like the ones in <span class="quarto-unresolved-ref">?fig-dice1</span> and <span class="quarto-unresolved-ref">?fig-dice2</span>.<br>
(As a sidenote: be sure to understand what the branch lengths mean in a phylogenetic tree. Here, these branch lengths denote the “distance” between two cells’ CNA profiles. (The longer the branch, the more it’s different.) Other methods use the branch length to denote time, i.e., the longer the branch, the more time has elapsed since the branching event. Most methods (in my experience) do not give you branch lengths relative to time, since this would require you (the data analyst) to provide the method with more biological information. For example, how fast do certain CNA events happen? You would need to provide this additional information to your phylogenetic method for it to know how to estimate branch lengths that denote time.)</p>
<div class="block">
<p><img src="fig/chap8/dice.png" id="fig-dice1" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From "> <img src="fig/chap8/medicc2.png" id="fig-dice2" class="img-fluid quarto-figure quarto-figure-center anchored" style="width:100.0%" alt="From "></p>
</div>
</section>
<section id="sec-dna_phylogenetic" class="level3" data-number="7.6.2">
<h3 data-number="7.6.2" class="anchored" data-anchor-id="sec-dna_phylogenetic"><span class="header-section-number">7.6.2</span> COMPASS, for scDNA copy number: Illustrating a probabilistic approach</h3>
<p>COMPASS (Joint Copy Number and Mutation Phylogeny from Amplicon Single-Cell DNA Sequencing) is a probabilistic model designed to reconstruct a branching evolutionary tree of SNVs (single-nucleotide variants) <em>and</em> CNAs (copy number alterations) from single-cell DNA sequencing data. It accommodates both allele-specific copy number changes (loss, gain, copy-neutral loss of heterozygosity) and single-nucleotide events, all integrated into a single phylogenetic structure.</p>
<p>(In COMPASS, there are meaningful branch lengths. The output is simply a cladogram.)</p>
<div class="block">
<div id="fig-compass-alg" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-compass-alg-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/compass_alg.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-compass-alg-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.15: From .<br>
Some basic terminology: 1) <strong>SNV (Single Nucleotide Variant):</strong> A change in a single nucleotide of the DNA sequence.<br>
2) <strong>Loss:</strong> (Not shown in this figure) A deletion of one copy of a chromosomal region. For instance, in the usual 23 pairs of human chromosomes, one region could be deleted from one copy of Chromosome 1, while remaining intact on the other copy.<br>
3) <strong>CNLOH (Copy-Neutral Loss of Heterozygosity):</strong> A copy number alteration that preserves the total number of chromosome copies but still removes heterozygosity. In other words, the cell ends up with the same number of copies overall, yet the two copies become identical (losing heterozygosity).<br>
4) <strong>Gain:</strong> The opposite of a “loss.” This could involve duplicating a specific chromosomal region (thus elongating that chromosome) or duplicating the entire chromosome (leading to three copies of Chromosome 1, instead of the usual two).
</figcaption>
</figure>
</div>
</div>
<div class="block3">
<p><strong>Input/Output.</strong> The input to COMPASS is<br>
1) <span class="math inline">\(D \in \mathbb{Z}_+^{n \times d}\)</span> denoting the total read count per DNA region (among <span class="math inline">\(d\)</span> regions) for each of the <span class="math inline">\(n\)</span> cells, and<br>
two matrices <span class="math inline">\(X, Y \in \mathbb{Z}_+^{n \times d}\)</span> denoting the number of DNA reads containing each SNV (among <span class="math inline">\(d\)</span> SNVs) among the <span class="math inline">\(n\)</span> cells.</p>
<p>For example, let <span class="math inline">\(X_{ij}\)</span> denote the number of DNA reads from cell <span class="math inline">\(i\)</span> that contained the major allele (i.e., “reference”) for SNV <span class="math inline">\(j\)</span>, and <span class="math inline">\(Y_{ij}\)</span> denote likewise but for the minor allele (i.e., “alternative”) for SNV <span class="math inline">\(j\)</span>.</p>
<p>The output is a cladogram among the <span class="math inline">\(n\)</span> cells. More specifically, COMPASS outputs a tree where the “internal nodes” denote what type of mutation happened, and cells are “attached” to these internal nodes (denoting that this cell followed this specific sequence of mutations, starting from the root of the tree).</p>
</div>
<p>Here are the overall ideas of COMPASS. Both total coverage (for copy number inference) and allele-specific coverage (for LOH and SNV calling) are going to be part of the likelihood to help model the likelihood of a particular tree (which will be used to determine the tree that best fits the data).</p>
<ul>
<li><p><strong>Model constraints</strong>:<br>
COMPASS models four classes of mutations:</p>
<ol type="1">
<li>SNVs, each occurring exactly once in the tree,<br>
</li>
<li>Loss, which removes one copy of an allele in a region,<br>
</li>
<li>Gain, which increases the copy number of a region, and<br>
</li>
<li>CNLOH (copy-neutral LOH), which replaces one allele with another while keeping total copy number constant.</li>
</ol>
<p>Each region can undergo CNAs at most once per lineage. When subclones exhibit multiple consecutive CNAs in a region, COMPASS in practice uses a single-event approximation to avoid ambiguous high copy numbers. (This is a parsimonious principle, in some sense. We’re dramatically limiting the number of trees that are going to be considered with these constraints.)</p></li>
<li><p><strong>Coverage and allelic likelihoods</strong>:<br>
There are two components to model the likelihood, one for the SNVs, one for the CNAs.</p>
<ul>
<li><p>For the CNAs, we have a <em>coverage model</em>. COMPASS uses a negative binomial distribution for each region’s total reads to capture overdispersion. Each region <span class="math inline">\(j \in \{1,\ldots,d\}\)</span> is assigned a baseline coverage weight <span class="math inline">\(\rho_k\)</span> (reflecting primer efficiency, etc.), so if a cell’s node in the phylogeny has copy number <span class="math inline">\(\text{cn}_k\)</span>, the expected read depth is scaled accordingly.</p></li>
<li><p>For the SNVs, we have an <em>allelic model</em>. For each locus <span class="math inline">\(j \in \{1,\ldots,p\}\)</span>, COMPASS directly uses read counts <span class="math inline">\((\text{REF}_i, \text{ALT}_i)\)</span> under a Beta-Binomial model, allowing for overdispersion in allele fraction. Each variant has a dropout rate <span class="math inline">\(\mu_i\)</span>, reflecting the probability that a given allele is lost (PCR failure), and the method marginalizes over all possible dropout configurations (<span class="math inline">\(k\)</span> reference copies vs.&nbsp;<span class="math inline">\(l\)</span> alternate copies).</p></li>
</ul></li>
<li><p><strong>Cell-to-node assignments</strong>:<br>
(Here, “node” refers to a mutation event, which are internal nodes in our tree.) COMPASS marginalizes over cell attachments to genotype nodes. Let <span class="math inline">\(\pi_k\)</span> be a prior probability that any cell is attached to node <span class="math inline">\(k\)</span>. The algorithm thus sums<br>
<span class="math display">\[
\sum_{k} \pi_k \, \prod_{i=1}^{p} P(\text{region reads}|\dots) \,\prod_{i=1}^{d} P(\text{allelic reads}|\dots)
\]</span><br>
for each cell.</p></li>
<li><p><strong>Simulated annealing</strong>:<br>
Simulated annealing is a useful statistical technique for reconstructing a phylogenetic tree from single-cell DNA data because we want to search through a vast space of possible evolutionary histories<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>. Each candidate tree is defined by how individual copy number events and mutations are arranged and passed down through successive subclones.</p>
<p>A naive or purely greedy search might easily get stuck in suboptimal solutions, given the noisy nature of single-cell data and the combinatorial explosion of possible trees. By gradually “cooling” the exploration of this space — starting from more random moves and then narrowing down to finer adjustments — simulated annealing helps ensure that the final solution is not merely a local optimum. In other words, this strategy balances the need for broad exploration early on with the requirement of a precise, fine-grained search later, thereby increasing the likelihood of discovering a high-probability lineage structure.</p></li>
</ul>
<p>COMPASS employs a simulated annealing approach over the discrete space of trees:</p>
<ol type="1">
<li>Start with a random initial tree (random parent pointers, random SNV allocation, no CNAs).</li>
<li>Propose local “moves” (adding or removing a node, reassigning mutations, introducing or removing a CNA, etc.).</li>
<li>Accept or reject based on the ratio of posterior probabilities at a current “temperature” to allow exploration; gradually decrease the temperature to refine solutions.</li>
</ol>
<p>To simulate potential trees, we need a prior distribution over trees. COMPASS will pick a prior to discourage trees with excessive branching or excessive CNA events. Specifically, it penalizes the total number of nodes and the number of CNA events. Longer CNAs across consecutive amplicons are treated as a single event. CNLOH events may be penalized more due to their strong effect on the allelic fraction.</p>
<p><strong>Parameter Updates.</strong> Node weights <span class="math inline">\(\{\pi_k\}\)</span> (i.e., the probability any cell is attached to an internal node) and other parameters are estimated by an Expectation-Maximization (EM) step:</p>
<ul>
<li><strong>E-step</strong>: Compute the posterior probabilities that cell <span class="math inline">\(i\)</span> belongs to node <span class="math inline">\(k\)</span>, given the current tree and parameters.</li>
<li><strong>M-step</strong>: Update <span class="math inline">\(\pi_k\)</span> as the average posterior membership across cells.</li>
</ul>
<p>This EM routine is invoked each time we propose or accept a new tree in the simulated annealing.<br>
(The EM algorithm is another useful statistical technique whenever you’re dealing with a model with unobserved variables. Here, our unobserved variable is whether or not a cell <span class="math inline">\(i\)</span> is attached to node <span class="math inline">\(k\)</span>.)</p>
<p>See <a href="#fig-compass" class="quarto-xref">Figure&nbsp;<span>7.16</span></a> for an illustration of what the results could look like.</p>
<div class="block">
<div id="fig-compass" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-compass-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/compass.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-compass-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.16: From .
</figcaption>
</figure>
</div>
</div>
</section>
<section id="a-brief-note-on-other-approaches-1" class="level3" data-number="7.6.3">
<h3 data-number="7.6.3" class="anchored" data-anchor-id="a-brief-note-on-other-approaches-1"><span class="header-section-number">7.6.3</span> A brief note on other approaches</h3>
<p>So… what exactly are these trees good for?<br>
As we’ve mentioned, you can see in <a href="#fig-compass" class="quarto-xref">Figure&nbsp;<span>7.16</span></a> that COMPASS is trying to learn the mutation event that happened at each node. (This gives us a clue on how deleterious the mutation event is. As a simple example, if there was a loss in copy number in a region that contains a cancer suppressive gene, we might expect rapid proliferation of this cancer cell.)</p>
<p>On the other hand, what we will talk more about in <a href="chapter9_spatial.qmd">Chapter 9</a> is combining these phylogenetic trees with spatial information about these cells. See <a href="#fig-cancer-phylogenetic-sc" class="quarto-xref">Figure&nbsp;<span>7.17</span></a>.</p>
<div class="block">
<div id="fig-cancer-phylogenetic-sc" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cancer-phylogenetic-sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="fig/chap8/cancer-phylogenetic-sc.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cancer-phylogenetic-sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7.17: From . Panel a is the copy number events for each cell (here, this is actually based on spatial transcriptomics), and panel b is the estimated phylogenetic tree. From the tree, they can look at the spatial position of each clone and literally see on the 2D tissue slice how cancer clones (potentially aggressively) expanded in this tissue slice. The phylogenetic information gave us clues on which cancer subclone originated from which.
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</figcaption>
</figure>
</div>
</div>


<<<<<<< HEAD
<<<<<<< HEAD
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-ashouri2023decoding" class="csl-entry" role="listitem">
Ashouri, Arghavan, Chufan Zhang, and Federico Gaiti. 2023. <span>“Decoding Cancer Evolution: Integrating Genetic and Non-Genetic Insights.”</span> <em>Genes</em> 14 (10): 1856.
</div>
<div id="ref-marine2020non" class="csl-entry" role="listitem">
Marine, Jean-Christophe, Sarah-Jane Dawson, and Mark A Dawson. 2020. <span>“Non-Genetic Mechanisms of Therapeutic Resistance in Cancer.”</span> <em>Nature Reviews Cancer</em> 20 (12): 743–56.
</div>
<div id="ref-stephenson2021single" class="csl-entry" role="listitem">
Stephenson, Emily, Gary Reynolds, Rachel A Botting, Fernando J Calero-Nieto, Michael D Morgan, Zewen Kelvin Tuong, Karsten Bach, et al. 2021. <span>“Single-Cell Multi-Omics Analysis of the Immune Response in COVID-19.”</span> <em>Nature Medicine</em> 27 (5): 904–16.
</div>
<div id="ref-weinreb2020lineage" class="csl-entry" role="listitem">
Weinreb, Caleb, Alejo Rodriguez-Fraticelli, Fernando D Camargo, and Allon M Klein. 2020. <span>“Lineage Tracing on Transcriptional Landscapes Links State to Fate During Differentiation.”</span> <em>Science</em> 367 (6479).
</div>
</div>
</section>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>See <a href="https://en.wikipedia.org/wiki/Simulated_annealing" class="uri">https://en.wikipedia.org/wiki/Simulated_annealing</a><a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>Formally, you would also need to prove that all the unique <span class="math inline">\(n\)</span>-leaf trees formed this way from one particular <span class="math inline">\((n-1)\)</span>-leaf tree is distinct from any possible <span class="math inline">\(n\)</span>-leaf tree formed from a different <span class="math inline">\((n-1)\)</span>-leaf tree.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>You can approximate <span class="math inline">\(n!\)</span> by <span class="math inline">\(n^n\)</span>. See <a href="https://proofwiki.org/wiki/Lower_and_Upper_Bound_o_Factorial" class="uri">https://proofwiki.org/wiki/Lower_and_Upper_Bound_o_Factorial</a> or <a href="https://en.wikipedia.org/wiki/Stirlings_approximation" class="uri">https://en.wikipedia.org/wiki/Stirlings_approximation</a>.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>See <a href="https://en.wikipedia.org/wiki/Simulated_annealing" class="uri">https://en.wikipedia.org/wiki/Simulated_annealing</a><a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>See <a href="https://en.wikipedia.org/wiki/Simulated_annealing" class="uri">https://en.wikipedia.org/wiki/Simulated_annealing</a><a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
<<<<<<< HEAD
<<<<<<< HEAD
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
<<<<<<< HEAD
<<<<<<< HEAD
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
=======
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
<<<<<<< HEAD
<<<<<<< HEAD
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./chapter7_crispr.html" class="pagination-link" aria-label="Single-cell CRISPR editting">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Single-cell CRISPR editting</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./chapter9_spatial.html" class="pagination-link" aria-label="'Single-cell' spatial transcriptomics">
        <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">‘Single-cell’ spatial transcriptomics</span></span> <i class="bi bi-arrow-right-short"></i>
=======
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./chapter5_epigenetics.html" class="pagination-link" aria-label="Single-cell epigenetics">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Single-cell epigenetics</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./references.html" class="pagination-link" aria-label="References">
        <span class="nav-page-text">References</span> <i class="bi bi-arrow-right-short"></i>
<<<<<<< HEAD
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
=======
>>>>>>> 84b7600d5159e099c748982f7f3d79cab9db2634
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>