<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>References – scOmicNotes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./chapter9_spatial.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./references.html">References</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">scOmicNotes</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Preface</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2_sequencing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Single-cell sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3_rna.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Single-cell RNA-sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4_protein.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">‘Single-cell’ proteomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5_epigenetics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Single-cell epigenetics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6_dna.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">‘Single-cell’ DNA</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7_crispr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Single-cell CRISPR editting</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8_lineage.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9_spatial.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">‘Single-cell’ spatial transcriptomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">References</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-adams2022single" class="csl-entry" role="listitem">
Adams, Levi, Min Kyung Song, Yoshiaki Tanaka, and Yoon-Seong Kim. 2022.
<span>“Single-Nuclei Paired Multiomic Analysis of Young, Aged, and
Parkinson’s Disease Human Midbrain Reveals Age-and Disease-Associated
Glial Changes and Their Contribution to Parkinson’s Disease.”</span>
<em>MedRxiv</em>, 2022–01.
</div>
<div id="ref-almeida2021employing" class="csl-entry" role="listitem">
Almeida, Nathalia, Matthew WH Chung, Elena M Drudi, Elise N Engquist,
Eva Hamrud, Abigail Isaacson, Victoria SK Tsang, Fiona M Watt, and
Francesca M Spagnoli. 2021. <span>“Employing Core Regulatory Circuits to
Define Cell Identity.”</span> <em>The EMBO Journal</em> 40 (10):
e106785.
</div>
<div id="ref-argelaguet2021computational" class="csl-entry" role="listitem">
Argelaguet, Ricard, Anna SE Cuomo, Oliver Stegle, and John C Marioni.
2021. <span>“Computational Principles and Challenges in Single-Cell Data
Integration.”</span> <em>Nature Biotechnology</em> 39 (10): 1202–15.
</div>
<div id="ref-ashuach2023multivi" class="csl-entry" role="listitem">
Ashuach, Tal, Mariano I Gabitto, Rohan V Koodli, Giuseppe-Antonio Saldi,
Michael I Jordan, and Nir Yosef. 2023. <span>“MultiVI: Deep Generative
Model for the Integration of Multimodal Data.”</span> <em>Nature
Methods</em>, 1–10.
</div>
<div id="ref-ashuach2022peakvi" class="csl-entry" role="listitem">
Ashuach, Tal, Daniel A Reidenbach, Adam Gayoso, and Nir Yosef. 2022.
<span>“PeakVI: A Deep Generative Model for Single-Cell Chromatin
Accessibility Analysis.”</span> <em>Cell Reports Methods</em> 2 (3).
</div>
<div id="ref-badia2023gene" class="csl-entry" role="listitem">
Badia-i-Mompel, Pau, Lorna Wessels, Sophia Müller-Dott, Rémi Trimbour,
Ricardo O Ramirez Flores, Ricard Argelaguet, and Julio Saez-Rodriguez.
2023. <span>“Gene Regulatory Network Inference in the Era of Single-Cell
Multi-Omics.”</span> <em>Nature Reviews Genetics</em> 24 (11): 739–54.
</div>
<div id="ref-blanco2020bivalent" class="csl-entry" role="listitem">
Blanco, Enrique, Mar González-Ramı́rez, Anna Alcaine-Colet, Sergi Aranda,
and Luciano Di Croce. 2020. <span>“The Bivalent Genome:
Characterization, Structure, and Regulation.”</span> <em>Trends in
Genetics</em> 36 (2): 118–31.
</div>
<div id="ref-bravo2023scenic" class="csl-entry" role="listitem">
Bravo González-Blas, Carmen, Seppe De Winter, Gert Hulselmans, Nikolai
Hecker, Irina Matetovici, Valerie Christiaens, Suresh Poovathingal,
Jasper Wouters, Sara Aibar, and Stein Aerts. 2023.
<span>“<span>SCENIC+</span>: Single-Cell Multiomic Inference of
Enhancers and Gene Regulatory Networks.”</span> <em>Nature Methods</em>
20 (9): 1355–67.
</div>
<div id="ref-bruggeman2008using" class="csl-entry" role="listitem">
Bruggeman-Everts, Fieke. 2008. <span>“Using Mouse Models to Identify the
Molecular Mechanisms Mediating Gene by Environment Interactions in
Depressive Illness.”</span> PhD thesis. <a href="https://www.researchgate.net/publication/233413663_Using_mouse_models_to_identify_the_molecular_mechanisms_mediating_gene_by_environment_interactions_in_depressive_illness">https://www.researchgate.net/publication/233413663_Using_mouse_models_to_identify_the_molecular_mechanisms_mediating_gene_by_environment_interactions_in_depressive_illness</a>.
</div>
<div id="ref-buenrostro2015atac" class="csl-entry" role="listitem">
Buenrostro, Jason D, Beijing Wu, Howard Y Chang, and William J
Greenleaf. 2015. <span>“ATAC-Seq: A Method for Assaying Chromatin
Accessibility Genome-Wide.”</span> <em>Current Protocols in Molecular
Biology</em> 109 (1): 21–29.
</div>
<div id="ref-cannoodt2021spearheading" class="csl-entry" role="listitem">
Cannoodt, Robrecht, Wouter Saelens, Louise Deconinck, and Yvan Saeys.
2021. <span>“Spearheading Future Omics Analyses Using Dyngen, a
Multi-Modal Simulator of Single Cells.”</span> <em>Nature
Communications</em> 12 (1): 3942.
</div>
<div id="ref-cao2022manifold" class="csl-entry" role="listitem">
Cao, Kai, Yiguang Hong, and Lin Wan. 2022. <span>“Manifold Alignment for
Heterogeneous Single-Cell Multi-Omics Data Integration Using
Pamona.”</span> <em>Bioinformatics</em> 38 (1): 211–19.
</div>
<div id="ref-cao2022multi" class="csl-entry" role="listitem">
Cao, Zhi-Jie, and Ge Gao. 2022. <span>“Multi-Omics Single-Cell Data
Integration and Regulatory Inference with Graph-Linked
Embedding.”</span> <em>Nature Biotechnology</em> 40 (10): 1458–66.
</div>
<div id="ref-cha2020single" class="csl-entry" role="listitem">
Cha, Junha, and Insuk Lee. 2020. <span>“Single-Cell Network Biology for
Resolving Cellular Heterogeneity in Human Diseases.”</span>
<em>Experimental &amp; Molecular Medicine</em> 52 (11): 1798–808.
</div>
<div id="ref-chen2024simba" class="csl-entry" role="listitem">
Chen, Huidong, Jayoung Ryu, Michael E Vinyard, Adam Lerer, and Luca
Pinello. 2024. <span>“SIMBA: Single-Cell Embedding Along with
Features.”</span> <em>Nature Methods</em> 21 (6): 1003–13.
</div>
<div id="ref-chen2015alternative" class="csl-entry" role="listitem">
Chen, J, and WA Weiss. 2015. <span>“Alternative Splicing in Cancer:
Implications for Biology and Therapy.”</span> <em>Oncogene</em> 34 (1):
1–14.
</div>
<div id="ref-chen2023integration" class="csl-entry" role="listitem">
Chen, Shuxiao, Bokai Zhu, Sijia Huang, John W Hickey, Kevin Z Lin,
Michael Snyder, William J Greenleaf, Garry P Nolan, Nancy R Zhang, and
Zongming Ma. 2024. <span>“Integration of Spatial and Single-Cell Data
Across Modalities with Weakly Linked Features.”</span> <em>Nature
Biotechnology</em> 42 (7): 1096–106.
</div>
<div id="ref-claringbould2021enhancers" class="csl-entry" role="listitem">
Claringbould, Annique, and Judith B Zaugg. 2021. <span>“Enhancers in
Disease: Molecular Basis and Emerging Treatment Strategies.”</span>
<em>Trends in Molecular Medicine</em> 27 (11): 1060–73.
</div>
<div id="ref-demetci2022scot" class="csl-entry" role="listitem">
Demetci, Pinar, Rebecca Santorella, Björn Sandstede, William Stafford
Noble, and Ritambhara Singh. 2022. <span>“SCOT: Single-Cell Multi-Omics
Alignment with Optimal Transport.”</span> <em>Journal of Computational
Biology</em> 29 (1): 3–18.
</div>
<div id="ref-ea2015contribution" class="csl-entry" role="listitem">
Ea, Vuthy, Marie-Odile Baudement, Annick Lesne, and Thierry Forné. 2015.
<span>“Contribution of Topological Domains and Loop Formation to 3D
Chromatin Organization.”</span> <em>Genes</em> 6 (3): 734–50.
</div>
<div id="ref-efron2002empirical" class="csl-entry" role="listitem">
Efron, Bradley, and Robert Tibshirani. 2002. <span>“Empirical Bayes
Methods and False Discovery Rates for Microarrays.”</span> <em>Genetic
Epidemiology</em> 23 (1): 70–86.
</div>
<div id="ref-fleck2023inferring" class="csl-entry" role="listitem">
Fleck, Jonas Simon, Sophie Martina Johanna Jansen, Damian Wollny, Fides
Zenk, Makiko Seimiya, Akanksha Jain, Ryoko Okamoto, et al. 2023.
<span>“Inferring and Perturbing Cell Fate Regulomes in Human Brain
Organoids.”</span> <em>Nature</em> 621 (7978): 365–72.
</div>
<div id="ref-fornes2020jaspar" class="csl-entry" role="listitem">
Fornes, Oriol, Jaime A Castro-Mondragon, Aziz Khan, Robin Van der Lee,
Xi Zhang, Phillip A Richmond, Bhavi P Modi, et al. 2020. <span>“JASPAR
2020: Update of the Open-Access Database of Transcription Factor Binding
Profiles.”</span> <em>Nucleic Acids Research</em> 48 (D1): D87–92.
</div>
<div id="ref-garbulowski2024genespider2" class="csl-entry" role="listitem">
Garbulowski, Mateusz, Thomas Hillerton, Daniel Morgan, Deniz Seçilmiş,
Lisbet Sonnhammer, Andreas Tjärnberg, Torbjörn EM Nordling, and Erik LL
Sonnhammer. 2024. <span>“GeneSPIDER2: Large Scale GRN Simulation and
Benchmarking with Perturbed Single-Cell Data.”</span> <em>NAR Genomics
and Bioinformatics</em> 6 (3): lqae121.
</div>
<div id="ref-gauba2021immunomodulation" class="csl-entry" role="listitem">
Gauba, Kavya, Shruti Gupta, Jyoti Shekhawat, Praveen Sharma, Dharmveer
Yadav, and Mithu Banerjee. 2021. <span>“Immunomodulation by Epigenome
Alterations in Mycobacterium Tuberculosis Infection.”</span>
<em>Tuberculosis</em> 128: 102077.
</div>
<div id="ref-glancy2024bivalent" class="csl-entry" role="listitem">
Glancy, Eleanor, Natalie Choy, and Melanie A Eckersley-Maslin. 2024.
<span>“Bivalent Chromatin: A Developmental Balancing Act Tipped in
Cancer.”</span> <em>Biochemical Society Transactions</em> 52 (1):
217–29.
</div>
<div id="ref-gohil2021applying" class="csl-entry" role="listitem">
Gohil, Satyen H, J Bryan Iorgulescu, David A Braun, Derin B Keskin, and
Kenneth J Livak. 2021. <span>“Applying High-Dimensional Single-Cell
Technologies to the Analysis of Cancer Immunotherapy.”</span> <em>Nature
Reviews Clinical Oncology</em> 18 (4): 244–56.
</div>
<div id="ref-golub1999molecular" class="csl-entry" role="listitem">
Golub, Todd R, Donna K Slonim, Pablo Tamayo, Christine Huard, Michelle
Gaasenbeek, Jill P Mesirov, Hilary Coller, et al. 1999. <span>“Molecular
Classification of Cancer: Class Discovery and Class Prediction by Gene
Expression Monitoring.”</span> <em>Science</em> 286 (5439): 531–37.
</div>
<div id="ref-gong2021cobolt" class="csl-entry" role="listitem">
Gong, Boying, Yun Zhou, and Elizabeth Purdom. 2021. <span>“Cobolt:
Integrative Analysis of Multimodal Single-Cell Sequencing Data.”</span>
<em>Genome Biology</em> 22: 1–21.
</div>
<div id="ref-gupta2022single" class="csl-entry" role="listitem">
Gupta, Chirag, Jielin Xu, Ting Jin, Saniya Khullar, Xiaoyu Liu, Sayali
Alatkar, Feixiong Cheng, and Daifeng Wang. 2022. <span>“Single-Cell
Network Biology Characterizes Cell Type Gene Regulation for Drug
Repurposing and Phenotype Prediction in Alzheimer’s Disease.”</span>
<em>PLOS Computational Biology</em> 18 (7): e1010287.
</div>
<div id="ref-habib2007analysis" class="csl-entry" role="listitem">
Habib, Naomi. 2007. <em>Analysis of DNA Motifs Based on a Novel Motif
Comparison Method</em>. Hebrew University of Jerusalem.
</div>
<div id="ref-hamma2006pseudouridine" class="csl-entry" role="listitem">
Hamma, Tomoko, and Adrian R Ferré-D’Amaré. 2006. <span>“Pseudouridine
Synthases.”</span> <em>Chemistry &amp; Biology</em> 13 (11): 1125–35.
</div>
<div id="ref-hansson2023blood" class="csl-entry" role="listitem">
Hansson, Oskar, Kaj Blennow, Henrik Zetterberg, and Jeffrey Dage. 2023.
<span>“Blood Biomarkers for Alzheimer’s Disease in Clinical Practice and
Trials.”</span> <em>Nature Aging</em> 3 (5): 506–19.
</div>
<div id="ref-hao2020integrated" class="csl-entry" role="listitem">
Hao, Yuhan, Stephanie Hao, Erica Andersen-Nissen, William M Mauck,
Shiwei Zheng, Andrew Butler, Maddie Jane Lee, et al. 2021.
<span>“Integrated Analysis of Multimodal Single-Cell Data.”</span>
<em>Cell</em> 184 (13): 3573–87.
</div>
<div id="ref-heinz2010simple" class="csl-entry" role="listitem">
Heinz, Sven, Christopher Benner, Nathanael Spann, Eric Bertolino, Yin C
Lin, Peter Laslo, Jason X Cheng, Cornelis Murre, Harinder Singh, and
Christopher K Glass. 2010. <span>“Simple Combinations of
Lineage-Determining Transcription Factors Prime Cis-Regulatory Elements
Required for Macrophage and b Cell Identities.”</span> <em>Molecular
Cell</em> 38 (4): 576–89.
</div>
<div id="ref-hu2024cellwalker2" class="csl-entry" role="listitem">
Hu, Zhirui, Pawel F Przytycki, and Katherine S Pollard. 2024.
<span>“CellWalker2: Multi-Omic Discovery of Hierarchical Cell Type
Relationships and Their Associations with Genomic Annotations.”</span>
<em>bioRxiv</em>.
</div>
<div id="ref-ito2022factors" class="csl-entry" role="listitem">
Ito, Shinsuke, Nando Dulal Das, Takashi Umehara, and Haruhiko Koseki.
2022. <span>“Factors and Mechanisms That Influence Chromatin-Mediated
Enhancer–Promoter Interactions and Transcriptional Regulation.”</span>
<em>Cancers</em> 14 (21): 5404.
</div>
<div id="ref-jiang2017scale" class="csl-entry" role="listitem">
Jiang, Yuchao, Nancy R Zhang, and Mingyao Li. 2017. <span>“SCALE:
Modeling Allele-Specific Gene Expression by Single-Cell RNA
Sequencing.”</span> <em>Genome Biology</em> 18: 1–15.
</div>
<div id="ref-karamveer2024approaches" class="csl-entry" role="listitem">
Karamveer, and Yasin Uzun. 2024. <span>“Approaches for Benchmarking
Single-Cell Gene Regulatory Network Methods.”</span> <em>Bioinformatics
and Biology Insights</em> 18: 11779322241287120.
</div>
<div id="ref-kepchia2020diverse" class="csl-entry" role="listitem">
Kepchia, Devin, Ling Huang, Richard Dargusch, Robert A Rissman, Maxim N
Shokhirev, Wolfgang Fischer, and David Schubert. 2020. <span>“Diverse
Proteins Aggregate in Mild Cognitive Impairment and Alzheimer’s Disease
Brain.”</span> <em>Alzheimer’s Research &amp; Therapy</em> 12: 1–20.
</div>
<div id="ref-kim2023gene" class="csl-entry" role="listitem">
Kim, Daniel, Andy Tran, Hani Jieun Kim, Yingxin Lin, Jean Yee Hwa Yang,
and Pengyi Yang. 2023. <span>“Gene Regulatory Network Reconstruction:
Harnessing the Power of Single-Cell Multi-Omic Data.”</span> <em>NPJ
Systems Biology and Applications</em> 9 (1): 51.
</div>
<div id="ref-leblanc2023major" class="csl-entry" role="listitem">
Leblanc, Francis JA, and Guillaume Lettre. 2023. <span>“Major Cell-Types
in Multiomic Single-Nucleus Datasets Impact Statistical Modeling of
Links Between Regulatory Sequences and Target Genes.”</span>
<em>Scientific Reports</em> 13 (1): 3924.
</div>
<div id="ref-lei2021applications" class="csl-entry" role="listitem">
Lei, Yalan, Rong Tang, Jin Xu, Wei Wang, Bo Zhang, Jiang Liu, Xianjun
Yu, and Si Shi. 2021. <span>“Applications of Single-Cell Sequencing in
Cancer Research: Progress and Perspectives.”</span> <em>Journal of
Hematology &amp; Oncology</em> 14 (1): 91.
</div>
<div id="ref-leiz2021technologies" class="csl-entry" role="listitem">
Leiz, Janna, Maria Rutkiewicz, Carmen Birchmeier, Udo Heinemann, and Kai
M Schmidt-Ott. 2021. <span>“Technologies for Profiling the Impact of
Genomic Variants on Transcription Factor Binding.”</span>
<em>Medizinische Genetik</em> 33 (2): 147–55.
</div>
<div id="ref-li2024dissecting" class="csl-entry" role="listitem">
Li, Jingyi Jessica, Heather J Zhou, Peter J Bickel, and Xin Tong. 2024.
<span>“Dissecting Gene Expression Heterogeneity: Generalized Pearson
Correlation Squares and the k-Lines Clustering Algorithm.”</span>
<em>Journal of the American Statistical Association</em>, 1–14.
</div>
<div id="ref-li2022population" class="csl-entry" role="listitem">
Li, Lei, Yumei Li, Xudong Zou, Fuduan Peng, Ya Cui, Eric J Wagner, and
Wei Li. 2022. <span>“Population-Scale Genetic Control of Alternative
Polyadenylation and Its Association with Human Diseases.”</span>
<em>Quantitative Biology</em> 10 (1): 44–54.
</div>
<div id="ref-lim2024advances" class="csl-entry" role="listitem">
Lim, Jongsu, Chanho Park, Minjae Kim, Hyukhee Kim, Junil Kim, and
Dong-Sung Lee. 2024. <span>“Advances in Single-Cell Omics and Multiomics
for High-Resolution Molecular Profiling.”</span> <em>Experimental &amp;
Molecular Medicine</em> 56 (3): 515–26.
</div>
<div id="ref-liu2021seeing" class="csl-entry" role="listitem">
Liu, Ning, Wai Yee Low, Hamid Alinejad-Rokny, Stephen Pederson, Timothy
Sadlon, Simon Barry, and James Breen. 2021. <span>“Seeing the Forest
Through the Trees: Prioritising Potentially Functional Interactions from
Hi-c.”</span> <em>Epigenetics &amp; Chromatin</em> 14 (1): 41.
</div>
<div id="ref-loers2024single" class="csl-entry" role="listitem">
Loers, Jens Uwe, and Vanessa Vermeirssen. 2024. <span>“A Single-Cell
Multimodal View on Gene Regulatory Network Inference from
Transcriptomics and Chromatin Accessibility Data.”</span> <em>Briefings
in Bioinformatics</em> 25 (5): bbae382.
</div>
<div id="ref-ma2020chromatin" class="csl-entry" role="listitem">
Ma, Sai, Bing Zhang, Lindsay M LaFave, Andrew S Earl, Zachary Chiang,
Yan Hu, Jiarui Ding, et al. 2020. <span>“Chromatin Potential Identified
by Shared Single-Cell Profiling of <span>RNA</span> and
Chromatin.”</span> <em>Cell</em> 183 (4): 1103–16.
</div>
<div id="ref-macrae2023regulation" class="csl-entry" role="listitem">
Macrae, Trisha A, Julie Fothergill-Robinson, and Miguel Ramalho-Santos.
2023. <span>“Regulation, Functions and Transmission of Bivalent
Chromatin During Mammalian Development.”</span> <em>Nature Reviews
Molecular Cell Biology</em> 24 (1): 6–26.
</div>
<div id="ref-mansisidor2022chromatin" class="csl-entry" role="listitem">
Mansisidor, Andrés R, and Viviana I Risca. 2022. <span>“Chromatin
Accessibility: Methods, Mechanisms, and Biological Insights.”</span>
<em>Nucleus</em> 13 (1): 238–78.
</div>
<div id="ref-martens2024modeling" class="csl-entry" role="listitem">
Martens, Laura D, David S Fischer, Vicente A Yépez, Fabian J Theis, and
Julien Gagneur. 2024. <span>“Modeling Fragment Counts Improves
Single-Cell ATAC-Seq Analysis.”</span> <em>Nature Methods</em> 21 (1):
28–31.
</div>
<div id="ref-mclean2010great" class="csl-entry" role="listitem">
McLean, Cory Y, Dave Bristor, Michael Hiller, Shoa L Clarke, Bruce T
Schaar, Craig B Lowe, Aaron M Wenger, and Gill Bejerano. 2010.
<span>“GREAT Improves Functional Interpretation of Cis-Regulatory
Regions.”</span> <em>Nature Biotechnology</em> 28 (5): 495–501.
</div>
<div id="ref-miao2021multi" class="csl-entry" role="listitem">
Miao, Zhen, Benjamin D Humphreys, Andrew P McMahon, and Junhyong Kim.
2021. <span>“Multiomics Integration in the Age of Million Single Cell
Data.”</span> <em>Nature Reviews Nephrology</em> 17 (11): 710–24.
</div>
<div id="ref-miao2024uniform" class="csl-entry" role="listitem">
Miao, Zhen, and Junhyong Kim. 2024. <span>“Uniform Quantification of
Single-Nucleus ATAC-Seq Data with Paired-Insertion Counting (PIC) and a
Model-Based Insertion Rate Estimator.”</span> <em>Nature Methods</em> 21
(1): 32–36.
</div>
<div id="ref-mignone2002untranslated" class="csl-entry" role="listitem">
Mignone, Flavio, Carmela Gissi, Sabino Liuni, and Graziano Pesole. 2002.
<span>“Untranslated Regions of mRNAs.”</span> <em>Genome Biology</em> 3:
1–10.
</div>
<div id="ref-moerman2019grnboost2" class="csl-entry" role="listitem">
Moerman, Thomas, Sara Aibar Santos, Carmen Bravo González-Blas, Jaak
Simm, Yves Moreau, Jan Aerts, and Stein Aerts. 2019. <span>“GRNBoost2
and Arboreto: Efficient and Scalable Inference of Gene Regulatory
Networks.”</span> <em>Bioinformatics</em> 35 (12): 2159–61.
</div>
<div id="ref-ng2021cascading" class="csl-entry" role="listitem">
Ng, Bernard, William Casazza, Nam Hee Kim, Chendi Wang, Farnush Farhadi,
Shinya Tasaki, David A Bennett, Philip L De Jager, Christopher Gaiteri,
and Sara Mostafavi. 2021. <span>“Cascading Epigenomic Analysis for
Identifying Disease Genes from the Regulatory Landscape of GWAS
Variants.”</span> <em>PLoS Genetics</em> 17 (11): e1009918.
</div>
<div id="ref-niazi2023rna" class="csl-entry" role="listitem">
Niazi, Sarfaraz K. 2023. <span>“RNA Therapeutics: A Healthcare Paradigm
Shift.”</span> <em>Biomedicines</em> 11 (5): 1275.
</div>
<div id="ref-pratapa2020benchmarking" class="csl-entry" role="listitem">
Pratapa, Aditya, Amogh P Jalihal, Jeffrey N Law, Aditya Bharadwaj, and
TM Murali. 2020. <span>“Benchmarking Algorithms for Gene Regulatory
Network Inference from Single-Cell Transcriptomic Data.”</span>
<em>Nature Methods</em> 17 (2): 147–54.
</div>
<div id="ref-rusilowicz2022protein" class="csl-entry" role="listitem">
Rusilowicz-Jones, Emma V, Sylvie Urbé, and Michael J Clague. 2022.
<span>“Protein Degradation on the Global Scale.”</span> <em>Molecular
Cell</em> 82 (8): 1414–23.
</div>
<div id="ref-schaff2024clonal" class="csl-entry" role="listitem">
Schaff, Dylan L, Aria J Fasse, Phoebe E White, Robert J Vander Velde,
and Sydney M Shaffer. 2024. <span>“Clonal Differences Underlie Variable
Responses to Sequential and Prolonged Treatment.”</span> <em>Cell
Systems</em> 15 (3): 213–26.
</div>
<div id="ref-schep2017chromvar" class="csl-entry" role="listitem">
Schep, Alicia N, Beijing Wu, Jason D Buenrostro, and William J
Greenleaf. 2017. <span>“<span class="nocase">chromVAR</span>: Inferring
Transcription-Factor-Associated Accessibility from Single-Cell
Epigenomic Data.”</span> <em>Nature Methods</em> 14 (10): 975–78.
</div>
<div id="ref-shalek2017single" class="csl-entry" role="listitem">
Shalek, Alex K, and Mikael Benson. 2017. <span>“Single-Cell Analyses to
Tailor Treatments.”</span> <em>Science Translational Medicine</em> 9
(408): eaan4730.
</div>
<div id="ref-siersbaek2014molecular" class="csl-entry" role="listitem">
Siersbæk, Rasmus, Songjoon Baek, Atefeh Rabiee, Ronni Nielsen, Sofie
Traynor, Nicholas Clark, Albin Sandelin, et al. 2014. <span>“Molecular
Architecture of Transcription Factor Hotspots in Early
Adipogenesis.”</span> <em>Cell Reports</em> 7 (5): 1434–42.
</div>
<div id="ref-da2021principles" class="csl-entry" role="listitem">
Silva, Paulo FL da, and Björn Schumacher. 2021. <span>“Principles of the
Molecular and Cellular Mechanisms of Aging.”</span> <em>Journal of
Investigative Dermatology</em> 141 (4): 951–60.
</div>
<div id="ref-steinbach2017pten" class="csl-entry" role="listitem">
Steinbach, Nicole. 2017. <em>PTEN Affects Gene Expression and Histone
Modifications and Plays a Role in the Regulation of Transcription</em>.
Columbia University.
</div>
<div id="ref-stock2024topological" class="csl-entry" role="listitem">
Stock, Marco, Niclas Popp, Jonathan Fiorentino, and Antonio Scialdone.
2024. <span>“Topological Benchmarking of Algorithms to Infer Gene
Regulatory Networks from Single-Cell RNA-Seq Data.”</span>
<em>Bioinformatics</em>, btae267.
</div>
<div id="ref-tang2023modelling" class="csl-entry" role="listitem">
Tang, Wenhao, Andreas Christ Sølvsten Jørgensen, Samuel Marguerat,
Philipp Thomas, and Vahid Shahrezaei. 2023. <span>“Modelling Capture
Efficiency of Single Cell RNA-Sequencing Data Improves Inference of
Transcriptome-Wide Burst Kinetics.”</span> <em>bioRxiv</em>, 2023–03.
</div>
<div id="ref-taylor2008apoptosis" class="csl-entry" role="listitem">
Taylor, Rebecca C, Sean P Cullen, and Seamus J Martin. 2008.
<span>“Apoptosis: Controlled Demolition at the Cellular Level.”</span>
<em>Nature Reviews Molecular Cell Biology</em> 9 (3): 231–41.
</div>
<div id="ref-teichmann2020method" class="csl-entry" role="listitem">
Teichmann, Sarah, and Mirjana Efremova. 2020. <span>“Method of the Year
2019: Single-Cell Multimodal Omics.”</span> <em>Nature Methods</em> 17
(1): 2020.
</div>
<div id="ref-tibshirani2005sparsity" class="csl-entry" role="listitem">
Tibshirani, Robert, Michael Saunders, Saharon Rosset, Ji Zhu, and Keith
Knight. 2005. <span>“Sparsity and Smoothness via the Fused
Lasso.”</span> <em>Journal of the Royal Statistical Society Series B:
Statistical Methodology</em> 67 (1): 91–108.
</div>
<div id="ref-verschoor2015introduction" class="csl-entry" role="listitem">
Verschoor, Chris P, Alina Lelic, Jonathan L Bramson, and Dawn ME
Bowdish. 2015. <span>“An Introduction to Automated Flow Cytometry Gating
Tools and Their Implementation.”</span> <em>Frontiers in Immunology</em>
6: 380.
</div>
<div id="ref-von2015microglial" class="csl-entry" role="listitem">
Von Bernhardi, Rommy, Laura Eugenı́n-von Bernhardi, and Jaime Eugenı́n.
2015. <span>“Microglial Cell Dysregulation in Brain Aging and
Neurodegeneration.”</span> <em>Frontiers in Aging Neuroscience</em> 7:
124.
</div>
<div id="ref-wang2023dictys" class="csl-entry" role="listitem">
Wang, Lingfei, Nikolaos Trasanidis, Ting Wu, Guanlan Dong, Michael Hu,
Daniel E Bauer, and Luca Pinello. 2023. <span>“Dictys: Dynamic Gene
Regulatory Network Dissects Developmental Continuum with Single-Cell
Multiomics.”</span> <em>Nature Methods</em> 20: 1368–78.
</div>
<div id="ref-wang2002precision" class="csl-entry" role="listitem">
Wang, Yulei, Chih Long Liu, John D Storey, Robert J Tibshirani, Daniel
Herschlag, and Patrick O Brown. 2002. <span>“Precision and Functional
Specificity in mRNA Decay.”</span> <em>Proceedings of the National
Academy of Sciences</em> 99 (9): 5860–65.
</div>
<div id="ref-wen2022single" class="csl-entry" role="listitem">
Wen, Lu, Guoqiang Li, Tao Huang, Wei Geng, Hao Pei, Jialiang Yang, Miao
Zhu, et al. 2022. <span>“Single-Cell Technologies: From Research to
Application.”</span> <em>The Innovation</em> 3 (6).
</div>
<div id="ref-wu2021babel" class="csl-entry" role="listitem">
Wu, Kevin E, Kathryn E Yost, Howard Y Chang, and James Zou. 2021.
<span>“<span>BABEL</span> Enables Cross-Modality Translation Between
Multiomic Profiles at Single-Cell Resolution.”</span> <em>Proceedings of
the National Academy of Sciences</em> 118 (15).
</div>
<div id="ref-xie2023robust" class="csl-entry" role="listitem">
Xie, Fangming, Ethan J Armand, Zizhen Yao, Hanqing Liu, Anna Bartlett, M
Margarita Behrens, Yang Eric Li, et al. 2023. <span>“Robust
Enhancer-Gene Regulation Identified by Single-Cell Transcriptomes and
Epigenomes.”</span> <em>Cell Genomics</em> 3 (7).
</div>
<div id="ref-xu2022diagonal" class="csl-entry" role="listitem">
Xu, Yang, and Rachel Patton McCord. 2022. <span>“Diagonal Integration of
Multimodal Single-Cell Data: Potential Pitfalls and Paths
Forward.”</span> <em>Nature Communications</em> 13 (1): 3505.
</div>
<div id="ref-yuan2024inferring" class="csl-entry" role="listitem">
Yuan, Qiuyue, and Zhana Duren. 2024. <span>“Inferring Gene Regulatory
Networks from Single-Cell Multiome Data Using Atlas-Scale External
Data.”</span> <em>Nature Biotechnology</em>, 1–11.
</div>
<div id="ref-zhang2024transcriptional" class="csl-entry" role="listitem">
Zhang, Qiuyu, Wenjie Cao, Jiaqi Wang, Yihao Yin, Rui Sun, Zunyi Tian,
Yuhan Hu, Yalan Tan, and Ben-gong Zhang. 2024. <span>“Transcriptional
Bursting Dynamics in Gene Expression.”</span> <em>Frontiers in
Genetics</em> 15: 1451461.
</div>
<div id="ref-zhang2022multimodal" class="csl-entry" role="listitem">
Zhang, Ran, Laetitia Meng-Papaxanthos, Jean-philippe Vert, and William
Stafford Noble. 2022. <span>“Multimodal Single-Cell Translation and
Alignment with Semi-Supervised Learning.”</span> <em>Journal of
Computational Biology</em> 29 (11): 1198–1212.
</div>
<div id="ref-zhang2021epigenetic" class="csl-entry" role="listitem">
Zhang, Ying, and Marc-André Sirard. 2021. <span>“Epigenetic Inheritance
of Acquired Traits Through DNA Methylation.”</span> <em>Animal
Frontiers</em> 11 (6): 19–27.
</div>
<div id="ref-zhang2023scmomat" class="csl-entry" role="listitem">
Zhang, Ziqi, Haoran Sun, Ragunathan Mariappan, Xi Chen, Xinyu Chen, Mika
S Jain, Mirjana Efremova, Sarah A Teichmann, Vaibhav Rajan, and Xiuwei
Zhang. 2023. <span>“scMoMaT Jointly Performs Single Cell Mosaic
Integration and Multi-Modal Bio-Marker Detection.”</span> <em>Nature
Communications</em> 14 (1): 384.
</div>
<div id="ref-zhang2022scdart" class="csl-entry" role="listitem">
Zhang, Ziqi, Chengkai Yang, and Xiuwei Zhang. 2022. <span>“scDART:
Integrating Unmatched scRNA-Seq and scATAC-Seq Data and Learning
Cross-Modality Relationship Simultaneously.”</span> <em>Genome
Biology</em> 23 (1): 139.
</div>
<div id="ref-zhou2024genetics" class="csl-entry" role="listitem">
Zhou, Yi, Hongjun Song, and Guo-li Ming. 2024. <span>“Genetics of Human
Brain Development.”</span> <em>Nature Reviews Genetics</em> 25 (1):
26–45.
</div>
<div id="ref-zhu2021robust" class="csl-entry" role="listitem">
Zhu, Bokai, Shuxiao Chen, Yunhao Bai, Han Chen, Nilanjan Mukherjee,
Gustavo Vazquez, David R McIlwain, et al. 2021. <span>“Robust
Single-Cell Matching and Multi-Modal Analysis Using Shared and Distinct
Features Reveals Orchestrated Immune Responses.”</span>
<em>bioRxiv</em>.
</div>
<div id="ref-von2022multiomic" class="csl-entry" role="listitem">
Ziegler, Lukas M von, Amalia Floriou-Servou, Rebecca Waag, Rebecca R Das
Gupta, Oliver Sturman, Katharina Gapp, Christina A Maat, et al. 2022.
<span>“Multiomic Profiling of the Acute Stress Response in the Mouse
Hippocampus.”</span> <em>Nature Communications</em> 13 (1): 1824.
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./chapter9_spatial.html" class="pagination-link" aria-label="'Single-cell' spatial transcriptomics">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">‘Single-cell’ spatial transcriptomics</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>