<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>References – scOmicNotes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./chapter9_spatial.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./references.html">References</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">scOmicNotes</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Preface</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter1_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter2_sequencing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Single-cell sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter3_rna.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Single-cell RNA-sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter4_protein.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">‘Single-cell’ proteomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter5_epigenetics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Single-cell epigenetics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter6_dna.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">‘Single-cell’ DNA</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter7_crispr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Single-cell CRISPR editting</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter8_lineage.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Single-cell lineage tracing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./chapter9_spatial.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">‘Single-cell’ spatial transcriptomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">References</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-ahlmann2023comparison" class="csl-entry" role="listitem">
Ahlmann-Eltze, Constantin, and Wolfgang Huber. 2023. <span>“Comparison
of Transformations for Single-Cell RNA-Seq Data.”</span> <em>Nature
Methods</em>, 1–8.
</div>
<div id="ref-arora2018analysis" class="csl-entry" role="listitem">
Arora, Sanjeev, Wei Hu, and Pravesh K Kothari. 2018. <span>“An Analysis
of the t-Sne Algorithm for Data Visualization.”</span> In <em>Conference
on Learning Theory</em>, 1455–62. PMLR.
</div>
<div id="ref-cai2022theoretical" class="csl-entry" role="listitem">
Cai, T Tony, and Rong Ma. 2022. <span>“Theoretical Foundations of t-Sne
for Visualizing High-Dimensional Clustered Data.”</span> <em>Journal of
Machine Learning Research</em> 23 (301): 1–54.
</div>
<div id="ref-carbonetto2021non" class="csl-entry" role="listitem">
Carbonetto, Peter, Abhishek Sarkar, Zihao Wang, and Matthew Stephens.
2021. <span>“Non-Negative Matrix Factorization Algorithms Greatly
Improve Topic Model Fits.”</span> <em>arXiv Preprint
arXiv:2105.13440</em>.
</div>
<div id="ref-chari2023specious" class="csl-entry" role="listitem">
Chari, Tara, and Lior Pachter. 2023. <span>“The Specious Art of
Single-Cell Genomics.”</span> <em>PLOS Computational Biology</em> 19
(8): e1011288.
</div>
<div id="ref-chen2018umi" class="csl-entry" role="listitem">
Chen, Wenan, Yan Li, John Easton, David Finkelstein, Gang Wu, and Xiang
Chen. 2018. <span>“<span>UMI</span>-Count Modeling and Differential
Expression Analysis for Single-Cell <span>RNA</span> Sequencing.”</span>
<em>Genome Biology</em> 19 (1): 70.
</div>
<div id="ref-claringbould2021enhancers" class="csl-entry" role="listitem">
Claringbould, Annique, and Judith B Zaugg. 2021. <span>“Enhancers in
Disease: Molecular Basis and Emerging Treatment Strategies.”</span>
<em>Trends in Molecular Medicine</em> 27 (11): 1060–73.
</div>
<div id="ref-efron2002empirical" class="csl-entry" role="listitem">
Efron, Bradley, and Robert Tibshirani. 2002. <span>“Empirical Bayes
Methods and False Discovery Rates for Microarrays.”</span> <em>Genetic
Epidemiology</em> 23 (1): 70–86.
</div>
<div id="ref-ghojogh2021factor" class="csl-entry" role="listitem">
Ghojogh, Benyamin, Ali Ghodsi, Fakhri Karray, and Mark Crowley. 2021.
<span>“Factor Analysis, Probabilistic Principal Component Analysis,
Variational Inference, and Variational Autoencoder: Tutorial and
Survey.”</span> <em>arXiv Preprint arXiv:2101.00734</em>.
</div>
<div id="ref-gohil2021applying" class="csl-entry" role="listitem">
Gohil, Satyen H, J Bryan Iorgulescu, David A Braun, Derin B Keskin, and
Kenneth J Livak. 2021. <span>“Applying High-Dimensional Single-Cell
Technologies to the Analysis of Cancer Immunotherapy.”</span> <em>Nature
Reviews Clinical Oncology</em> 18 (4): 244–56.
</div>
<div id="ref-golub1999molecular" class="csl-entry" role="listitem">
Golub, Todd R, Donna K Slonim, Pablo Tamayo, Christine Huard, Michelle
Gaasenbeek, Jill P Mesirov, Hilary Coller, et al. 1999. <span>“Molecular
Classification of Cancer: Class Discovery and Class Prediction by Gene
Expression Monitoring.”</span> <em>Science</em> 286 (5439): 531–37.
</div>
<div id="ref-hafemeister2019normalization" class="csl-entry" role="listitem">
Hafemeister, Christoph, and Rahul Satija. 2019. <span>“Normalization and
Variance Stabilization of Single-Cell <span class="nocase">RNA-seq</span> Data Using Regularized Negative Binomial
Regression.”</span> <em>Genome Biology</em> 20 (1): 1–15.
</div>
<div id="ref-haghverdi2015diffusion" class="csl-entry" role="listitem">
Haghverdi, Laleh, Florian Buettner, and Fabian J Theis. 2015.
<span>“Diffusion Maps for High-Dimensional Single-Cell Analysis of
Differentiation Data.”</span> <em>Bioinformatics</em> 31 (18): 2989–98.
</div>
<div id="ref-hansson2023blood" class="csl-entry" role="listitem">
Hansson, Oskar, Kaj Blennow, Henrik Zetterberg, and Jeffrey Dage. 2023.
<span>“Blood Biomarkers for Alzheimer’s Disease in Clinical Practice and
Trials.”</span> <em>Nature Aging</em> 3 (5): 506–19.
</div>
<div id="ref-heumos2023best" class="csl-entry" role="listitem">
Heumos, Lukas, Anna C Schaar, Christopher Lance, Anastasia Litinetskaya,
Felix Drost, Luke Zappia, Malte D Lücken, et al. 2023. <span>“Best
Practices for Single-Cell Analysis Across Modalities.”</span> <em>Nature
Reviews Genetics</em> 24 (8): 550–72.
</div>
<div id="ref-hou2020systematic" class="csl-entry" role="listitem">
Hou, Wenpin, Zhicheng Ji, Hongkai Ji, and Stephanie C Hicks. 2020.
<span>“A Systematic Evaluation of Single-Cell
<span>RNA</span>-Sequencing Imputation Methods.”</span> <em>Genome
Biology</em> 21 (1): 1–30.
</div>
<div id="ref-huang2022towards" class="csl-entry" role="listitem">
Huang, Haiyang, Yingfan Wang, Cynthia Rudin, and Edward P Browne. 2022.
<span>“Towards a Comprehensive Evaluation of Dimension Reduction Methods
for Transcriptomic Data Visualization.”</span> <em>Communications
Biology</em> 5 (1): 719.
</div>
<div id="ref-ito2022factors" class="csl-entry" role="listitem">
Ito, Shinsuke, Nando Dulal Das, Takashi Umehara, and Haruhiko Koseki.
2022. <span>“Factors and Mechanisms That Influence Chromatin-Mediated
Enhancer–Promoter Interactions and Transcriptional Regulation.”</span>
<em>Cancers</em> 14 (21): 5404.
</div>
<div id="ref-johnson2022embedr" class="csl-entry" role="listitem">
Johnson, Eric M, William Kath, and Madhav Mani. 2022.
<span>“<span>EMBEDR</span>: Distinguishing Signal from Noise in
Single-Cell Omics Data.”</span> <em>Patterns</em> 3 (3).
</div>
<div id="ref-kent1979multivariate" class="csl-entry" role="listitem">
Kent, JT, John Bibby, and KV Mardia. 1979. <em>Multivariate
Analysis</em>. Academic Press Amsterdam.
</div>
<div id="ref-kepchia2020diverse" class="csl-entry" role="listitem">
Kepchia, Devin, Ling Huang, Richard Dargusch, Robert A Rissman, Maxim N
Shokhirev, Wolfgang Fischer, and David Schubert. 2020. <span>“Diverse
Proteins Aggregate in Mild Cognitive Impairment and Alzheimer’s Disease
Brain.”</span> <em>Alzheimer’s Research &amp; Therapy</em> 12: 1–20.
</div>
<div id="ref-kharchenko2014bayesian" class="csl-entry" role="listitem">
Kharchenko, Peter V, Lev Silberstein, and David T Scadden. 2014.
<span>“Bayesian Approach to Single-Cell Differential Expression
Analysis.”</span> <em>Nature Methods</em> 11 (7): 740.
</div>
<div id="ref-korsunsky2019fast" class="csl-entry" role="listitem">
Korsunsky, Ilya, Nghia Millard, Jean Fan, Kamil Slowikowski, Fan Zhang,
Kevin Wei, Yuriy Baglaenko, Michael Brenner, Po-ru Loh, and Soumya
Raychaudhuri. 2019. <span>“Fast, Sensitive and Accurate Integration of
Single-Cell Data with Harmony.”</span> <em>Nature Methods</em>, 1–8.
</div>
<div id="ref-lause2021analytic" class="csl-entry" role="listitem">
Lause, Jan, Philipp Berens, and Dmitry Kobak. 2021. <span>“Analytic
Pearson Residuals for Normalization of Single-Cell RNA-Seq UMI
Data.”</span> <em>Genome Biology</em> 22: 1–20.
</div>
<div id="ref-lause2024art" class="csl-entry" role="listitem">
———. 2024. <span>“The Art of Seeing the Elephant in the Room: 2D
Embeddings of Single-Cell Data Do Make Sense.”</span> <em>bioRxiv</em>.
</div>
<div id="ref-lei2021applications" class="csl-entry" role="listitem">
Lei, Yalan, Rong Tang, Jin Xu, Wei Wang, Bo Zhang, Jiang Liu, Xianjun
Yu, and Si Shi. 2021. <span>“Applications of Single-Cell Sequencing in
Cancer Research: Progress and Perspectives.”</span> <em>Journal of
Hematology &amp; Oncology</em> 14 (1): 91.
</div>
<div id="ref-li2022population" class="csl-entry" role="listitem">
Li, Lei, Yumei Li, Xudong Zou, Fuduan Peng, Ya Cui, Eric J Wagner, and
Wei Li. 2022. <span>“Population-Scale Genetic Control of Alternative
Polyadenylation and Its Association with Human Diseases.”</span>
<em>Quantitative Biology</em> 10 (1): 44–54.
</div>
<div id="ref-lin2022esvd" class="csl-entry" role="listitem">
Lin, Kevin Z, Yixuan Qiu, and Kathryn Roeder. 2022. <span>“<span class="nocase">eSVD</span>: Cohort-Level Differential Expression in
Multi-Individual Single-Cell <span>RNA</span>-Seq Data Using
Exponential-Family Embeddings.”</span> <em>(In Preparation)</em>.
</div>
<div id="ref-lin2023esvd" class="csl-entry" role="listitem">
———. 2024. <span>“<span class="nocase">eSVD-DE</span>: Cohort-Wide
Differential Expression in Single-Cell <span>RNA</span>-Seq Data Using
Exponential-Family Embeddings.”</span> <em>BMC Bioinformatics</em> 25
(1): 113.
</div>
<div id="ref-lopez2018deep" class="csl-entry" role="listitem">
Lopez, Romain, Jeffrey Regier, Michael B Cole, Michael I Jordan, and Nir
Yosef. 2018. <span>“Deep Generative Modeling for Single-Cell
Transcriptomics.”</span> <em>Nature Methods</em> 15 (12): 1053.
</div>
<div id="ref-love2014moderated" class="csl-entry" role="listitem">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014.
<span>“Moderated Estimation of Fold Change and Dispersion for
<span>RNA</span>-Seq Data with <span>DESeq2</span>.”</span> <em>Genome
Biology</em> 15 (12): 550.
</div>
<div id="ref-luecken2022benchmarking" class="csl-entry" role="listitem">
Luecken, Malte D, Maren Büttner, Kridsadakorn Chaichoompu, Anna Danese,
Marta Interlandi, Michaela F Müller, Daniel C Strobl, et al. 2022.
<span>“Benchmarking Atlas-Level Data Integration in Single-Cell
Genomics.”</span> <em>Nature Methods</em> 19 (1): 41–50.
</div>
<div id="ref-luecken2019current" class="csl-entry" role="listitem">
Luecken, Malte D, and Fabian J Theis. 2019. <span>“Current Best
Practices in Single-Cell <span>RNA</span>-Seq Analysis: A
Tutorial.”</span> <em>Molecular Systems Biology</em> 15 (6): e8746.
</div>
<div id="ref-ma2024principled" class="csl-entry" role="listitem">
Ma, Rong, Eric D Sun, David Donoho, and James Zou. 2024.
<span>“Principled and Interpretable Alignability Testing and Integration
of Single-Cell Data.”</span> <em>Proceedings of the National Academy of
Sciences</em> 121 (10): e2313719121.
</div>
<div id="ref-maden2023challenges" class="csl-entry" role="listitem">
Maden, Sean K, Sang Ho Kwon, Louise A Huuki-Myers, Leonardo
Collado-Torres, Stephanie C Hicks, and Kristen R Maynard. 2023.
<span>“Challenges and Opportunities to Computationally Deconvolve
Heterogeneous Tissue with Varying Cell Sizes Using Single-Cell
RNA-Sequencing Datasets.”</span> <em>Genome Biology</em> 24 (1): 288.
</div>
<div id="ref-mcginnis2019doubletfinder" class="csl-entry" role="listitem">
McGinnis, Christopher S, Lyndsay M Murrow, and Zev J Gartner. 2019.
<span>“DoubletFinder: Doublet Detection in Single-Cell RNA Sequencing
Data Using Artificial Nearest Neighbors.”</span> <em>Cell Systems</em> 8
(4): 329–37.
</div>
<div id="ref-prater2024all" class="csl-entry" role="listitem">
Prater, Katherine E, and Kevin Z Lin. 2024. <span>“All the Single Cells:
Single-Cell Transcriptomics/Epigenomics Experimental Design and Analysis
Considerations for Glial Biologists.”</span> <em>Glia</em>.
</div>
<div id="ref-qian2022scinsight" class="csl-entry" role="listitem">
Qian, Kun, Shiwei Fu, Hongwei Li, and Wei Vivian Li. 2022.
<span>“scINSIGHT for Interpreting Single-Cell Gene Expression from
Biologically Heterogeneous Data.”</span> <em>Genome Biology</em> 23 (1):
1–23.
</div>
<div id="ref-risso2014normalization" class="csl-entry" role="listitem">
Risso, Davide, John Ngai, Terence P Speed, and Sandrine Dudoit. 2014.
<span>“Normalization of <span>RNA</span>-Seq Data Using Factor Analysis
of Control Genes or Samples.”</span> <em>Nature Biotechnology</em> 32
(9): 896–902.
</div>
<div id="ref-salzberg2018open" class="csl-entry" role="listitem">
Salzberg, Steven L. 2018. <span>“Open Questions: How Many Genes Do We
Have?”</span> <em>BMC Biology</em> 16 (1): 94.
</div>
<div id="ref-sarkar2021separating" class="csl-entry" role="listitem">
Sarkar, Abhishek, and Matthew Stephens. 2021. <span>“Separating
Measurement and Expression Models Clarifies Confusion in Single Cell
<span>RNA</span>-Seq Analysis.”</span> <em>Nature Genetics</em> 53 (6):
770–77.
</div>
<div id="ref-saunders2023embryo" class="csl-entry" role="listitem">
Saunders, Lauren M, Sanjay R Srivatsan, Madeleine Duran, Michael W
Dorrity, Brent Ewing, Tor H Linbo, Jay Shendure, et al. 2023.
<span>“Embryo-Scale Reverse Genetics at Single-Cell Resolution.”</span>
<em>Nature</em> 623 (7988): 782–91.
</div>
<div id="ref-schaff2024clonal" class="csl-entry" role="listitem">
Schaff, Dylan L, Aria J Fasse, Phoebe E White, Robert J Vander Velde,
and Sydney M Shaffer. 2024. <span>“Clonal Differences Underlie Variable
Responses to Sequential and Prolonged Treatment.”</span> <em>Cell
Systems</em> 15 (3): 213–26.
</div>
<div id="ref-setty2019characterization" class="csl-entry" role="listitem">
Setty, Manu, Vaidotas Kiseliovas, Jacob Levine, Adam Gayoso, Linas
Mazutis, and Dana Pe’er. 2019. <span>“Characterization of Cell Fate
Probabilities in Single-Cell Data with <span>Palantir</span>.”</span>
<em>Nature Biotechnology</em> 37 (4): 451–60.
</div>
<div id="ref-shalek2017single" class="csl-entry" role="listitem">
Shalek, Alex K, and Mikael Benson. 2017. <span>“Single-Cell Analyses to
Tailor Treatments.”</span> <em>Science Translational Medicine</em> 9
(408): eaan4730.
</div>
<div id="ref-da2021principles" class="csl-entry" role="listitem">
Silva, Paulo FL da, and Björn Schumacher. 2021. <span>“Principles of the
Molecular and Cellular Mechanisms of Aging.”</span> <em>Journal of
Investigative Dermatology</em> 141 (4): 951–60.
</div>
<div id="ref-teichmann2020method" class="csl-entry" role="listitem">
Teichmann, Sarah, and Mirjana Efremova. 2020. <span>“Method of the Year
2019: Single-Cell Multimodal Omics.”</span> <em>Nature Methods</em> 17
(1): 2020.
</div>
<div id="ref-tibshirani2005sparsity" class="csl-entry" role="listitem">
Tibshirani, Robert, Michael Saunders, Saharon Rosset, Ji Zhu, and Keith
Knight. 2005. <span>“Sparsity and Smoothness via the Fused
Lasso.”</span> <em>Journal of the Royal Statistical Society Series B:
Statistical Methodology</em> 67 (1): 91–108.
</div>
<div id="ref-tipping1999probabilistic" class="csl-entry" role="listitem">
Tipping, Michael E, and Christopher M Bishop. 1999. <span>“Probabilistic
Principal Component Analysis.”</span> <em>Journal of the Royal
Statistical Society: Series B (Statistical Methodology)</em> 61 (3):
611–22.
</div>
<div id="ref-townes2019feature" class="csl-entry" role="listitem">
Townes, F William, Stephanie C Hicks, Martin J Aryee, and Rafael A
Irizarry. 2019. <span>“Feature Selection and Dimension Reduction for
Single-Cell <span class="nocase">RNA-seq</span> Based on a Multinomial
Model.”</span> <em>Genome Biology</em> 20 (1): 1–16.
</div>
<div id="ref-tran2020benchmark" class="csl-entry" role="listitem">
Tran, Hoa Thi Nhu, Kok Siong Ang, Marion Chevrier, Xiaomeng Zhang,
Nicole Yee Shin Lee, Michelle Goh, and Jinmiao Chen. 2020. <span>“A
Benchmark of Batch-Effect Correction Methods for Single-Cell
<span>RNA</span> Sequencing Data.”</span> <em>Genome Biology</em> 21
(1): 1–32.
</div>
<div id="ref-van2018recovering" class="csl-entry" role="listitem">
Van Dijk, David, Roshan Sharma, Juozas Nainys, Kristina Yim, Pooja
Kathail, Ambrose J Carr, Cassandra Burdziak, et al. 2018.
<span>“Recovering Gene Interactions from Single-Cell Data Using Data
Diffusion.”</span> <em>Cell</em> 174 (3): 716–29.
</div>
<div id="ref-verschoor2015introduction" class="csl-entry" role="listitem">
Verschoor, Chris P, Alina Lelic, Jonathan L Bramson, and Dawn ME
Bowdish. 2015. <span>“An Introduction to Automated Flow Cytometry Gating
Tools and Their Implementation.”</span> <em>Frontiers in Immunology</em>
6: 380.
</div>
<div id="ref-von2015microglial" class="csl-entry" role="listitem">
Von Bernhardi, Rommy, Laura Eugenı́n-von Bernhardi, and Jaime Eugenı́n.
2015. <span>“Microglial Cell Dysregulation in Brain Aging and
Neurodegeneration.”</span> <em>Frontiers in Aging Neuroscience</em> 7:
124.
</div>
<div id="ref-wei2019genome" class="csl-entry" role="listitem">
Wei, Lai, Derek Lee, Cheuk-Ting Law, Misty Shuo Zhang, Jialing Shen, Don
Wai-Ching Chin, Allen Zhang, et al. 2019. <span>“Genome-Wide CRISPR/Cas9
Library Screening Identified PHGDH as a Critical Driver for Sorafenib
Resistance in HCC.”</span> <em>Nature Communications</em> 10 (1): 4681.
</div>
<div id="ref-wen2022single" class="csl-entry" role="listitem">
Wen, Lu, Guoqiang Li, Tao Huang, Wei Geng, Hao Pei, Jialiang Yang, Miao
Zhu, et al. 2022. <span>“Single-Cell Technologies: From Research to
Application.”</span> <em>The Innovation</em> 3 (6).
</div>
<div id="ref-xi2021benchmarking" class="csl-entry" role="listitem">
Xi, Nan Miles, and Jingyi Jessica Li. 2021. <span>“Benchmarking
Computational Doublet-Detection Methods for Single-Cell RNA Sequencing
Data.”</span> <em>Cell Systems</em> 12 (2): 176–94.
</div>
<div id="ref-xia2024statistical" class="csl-entry" role="listitem">
Xia, Lucy, Christy Lee, and Jingyi Jessica Li. 2024. <span>“Statistical
Method scDEED for Detecting Dubious 2D Single-Cell Embeddings and
Optimizing t-SNE and UMAP Hyperparameters.”</span> <em>Nature
Communications</em> 15 (1): 1753.
</div>
<div id="ref-yazar2022single" class="csl-entry" role="listitem">
Yazar, Seyhan, Jose Alquicira-Hernandez, Kristof Wing, Anne Senabouth, M
Grace Gordon, Stacey Andersen, Qinyi Lu, et al. 2022. <span>“Single-Cell
<span class="nocase">eQTL</span> Mapping Identifies Cell Type-Specific
Genetic Control of Autoimmune Disease.”</span> <em>Science</em> 376
(6589): eabf3041.
</div>
<div id="ref-young2020soupx" class="csl-entry" role="listitem">
Young, Matthew D, and Sam Behjati. 2020. <span>“SoupX Removes Ambient
RNA Contamination from Droplet-Based Single-Cell RNA Sequencing
Data.”</span> <em>Gigascience</em> 9 (12): giaa151.
</div>
<div id="ref-zhang2020combat" class="csl-entry" role="listitem">
Zhang, Yuqing, Giovanni Parmigiani, and W Evan Johnson. 2020.
<span>“ComBat-Seq: Batch Effect Adjustment for RNA-Seq Count
Data.”</span> <em>NAR Genomics and Bioinformatics</em> 2 (3): lqaa078.
</div>
<div id="ref-zhao2024systematic" class="csl-entry" role="listitem">
Zhao, Ruzhang, Jiuyao Lu, Weiqiang Zhou, Ni Zhao, and Hongkai Ji. 2024.
<span>“A Systematic Evaluation of Highly Variable Gene Selection Methods
for Single-Cell RNA-Sequencing.”</span> <em>bioRxiv</em>, 2024–08.
</div>
<div id="ref-zhou2024genetics" class="csl-entry" role="listitem">
Zhou, Yi, Hongjun Song, and Guo-li Ming. 2024. <span>“Genetics of Human
Brain Development.”</span> <em>Nature Reviews Genetics</em> 25 (1):
26–45.
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./chapter9_spatial.html" class="pagination-link" aria-label="'Single-cell' spatial transcriptomics">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">‘Single-cell’ spatial transcriptomics</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>